---
title: "TPP hit calling and evaluation"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide  
---

# General settings


```{r setup}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  include = TRUE,
  cache = TRUE,
  cache.lazy = FALSE,
  eval = TRUE
  # fig.width = 6*(1+sqrt(5))/2, fig.height = 6 #golden ratio
)

knitr::opts_knit$set(root.dir = "C:/Users/burtsche/Documents/COSMOS-TPP_paper")
```

## Load packages

```{r, message=F, warning =F}
library(tidyverse)
library("reshape2")
library(OmnipathR)
#library(xlsx)
library(viridis)
library(ggrepel)
options(connectionObserver = NULL)
library(org.Hs.eg.db)
library(visNetwork)
library(knitr)
library(UpSetR)
library(TPP2D)
library(cowplot)
library(ReactomePA)

mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
```

## Functions

```{r RPA function, message=F, warning =F}
GSE_RPA <- function(geneList, universe) {
  geneList <- mapIds(org.Hs.eg.db, geneList, "ENTREZID", "SYMBOL")
  universe <- mapIds(org.Hs.eg.db, universe,'ENTREZID','SYMBOL')
  pathway_enrichment <- enrichPathway(
    gene = geneList,
    organism = "human",
    pvalueCutoff = 1,
    pAdjustMethod = "BH",
    qvalueCutoff = 1,
    universe = universe,
    minGSSize = 5,
    maxGSSize = 500,
    readable = TRUE
  )
}
```


```{r}
values_effect <- c(
  "stabilized" = "darkcyan",
  "destabilized" = "aquamarine3",
  "increased abundance" = "maroon",
  "decreased abundance" = "salmon"
)
```

## Load data

```{r, message=F, warning=F}
TPP2D_UWB <- read.table("data/2DTPP_UWB_Ola_2021-03-24.txt", row.names = 1, sep = "\t")

load("data/220531_limma_multiomics_forpaper.RData")
load("data/KSN.RData")
load("data/210802_limma_UWB1.289_initialdataset_correctedphospho.RData")
load("data/220510_viper_footprints.RData")

load("data/KSN.RData")
load("data/dorothea.RData")
```

Aggregate and filter TPP UWB data to considered condition and quality criteria.

```{r}
proteins_merged <- TPP2D_UWB %>%
  filter(condition == "Olaparib 24h") %>%
  # convert temperature to numeric
  mutate(
    qupm = as.numeric(qupm),
    temperature = as.numeric(gsub("C", "", temperature)),
    gene_name = paste(representative, gene_name, sep = ".")
  ) %>%
  # filter proteins which were found with qupm > 1 at least once
  group_by(gene_name) %>%
  filter(any(qupm > 1)) %>%
  ungroup() %>%
  # extract relevant FCs
  dplyr::select(gene_name, matches("rel_fc_protein_.+uM$"), temperature, hit_call) %>%
  melt(id.vars = c("gene_name", "temperature", "hit_call")) %>%
  mutate(concentration = as.numeric(sub("^.*_(.+)uM$", "\\1", variable))) %>%
  # convert FC to log2fc
  mutate(log2_value = log2(value))
```

# TPP hit calling

The permutation takes one night!!! Don't run this chunk per default (eval=F).

```{r, eval = F}
TPPstatistical_input <- list()
for (file in c(
  "TPPraw_uwb_ola24_48.csv", "TPPraw_uwb_ola24_49.csv", "TPPraw_uwb_ola24_50.csv",
  "TPPraw_uwb_ola24_51.csv", "TPPraw_uwb_ola24_52.csv", "TPPraw_uwb_ola24_53.csv"
)) {
  tmp_file <- read_csv(paste0("data/raw/", file))
  tmp_file <- tmp_file %>%
    mutate(
      gene_name = `Gene name`, # paste(`Gene name`, Accession, sep = "."),
      protein_id = Accession
    ) %>% # c("1": as.character(nrow(tmp_file))),
    # protein_id = as.character(protein_id))  %>%
    dplyr::select(protein_id, gene_name, QUPM, matches("sumionarea"), matches("fold change")) %>%
    setNames(c(
      "protein_id", "gene_name", "qupm", "signal_sum_126", "signal_sum_127L", "signal_sum_127H", "signal_sum_128L", "signal_sum_128H",
      "signal_sum_129L", "signal_sum_129H", "signal_sum_130L", "signal_sum_130H", "signal_sum_131L", "rel_fc_126", "rel_fc_127L", "rel_fc_127H",
      "rel_fc_128L", "rel_fc_128H", "rel_fc_129L", "rel_fc_129H", "rel_fc_130L", "rel_fc_130H", "rel_fc_131L"
    )) %>%
    mutate_at(.funs = ~ . / rel_fc_128H, .vars = c("rel_fc_126", "rel_fc_127L", "rel_fc_127H", "rel_fc_128L", "rel_fc_128H"))

  TPPstatistical_input <- append(TPPstatistical_input, list(tmp_file))
}


data("config_tab")
data("raw_dat_list")

names(TPPstatistical_input) <- c("exp1", "exp2", "exp3", "exp4", "exp5", "exp6")

configuration <- config_tab %>%
  mutate(
    Compound = "Olaparib",
    Temperature = c(42.1, 44.1, 46.2, 48.1, 50.4, 51.9, 54, 56.1, 58.2, 60.1, 62.4, 63.9),
    RefCol = c("128H", "131L", "128H", "131L", "128H", "131L", "128H", "131L", "128H", "131L", "128H", "131L"),
    `126` = c("10", "-", "10", "-", "10", "-", "10", "-", "10", "-", "10", "-"),
    `127L` = c("4", "-", "4", "-", "4", "-", "4", "-", "4", "-", "4", "-"),
    `127H` = c("1", "-", "1", "-", "1", "-", "1", "-", "1", "-", "1", "-"),
    `128L` = c("0.4", "-", "0.4", "-", "0.4", "-", "0.4", "-", "0.4", "-", "0.4", "-"),
    `128H` = c("0", "-", "0", "-", "0", "-", "0", "-", "0", "-", "0", "-"),
    `129L` = c("-", "10", "-", "10", "-", "10", "-", "10", "-", "10", "-", "10"),
    `129H` = c("-", "4", "-", "4", "-", "4", "-", "4", "-", "4", "-", "4"),
    `130L` = c("-", "1", "-", "1", "-", "1", "-", "1", "-", "1", "-", "1"),
    `130H` = c("-", "0.4", "-", "0.4", "-", "0.4", "-", "0.4", "-", "0.4", "-", "0.4"),
    `131L` = c("-", "0", "-", "0", "-", "0", "-", "0", "-", "0", "-", "0")
  )

import_df <- import2dDataset(
  configTable = configuration,
  data = TPPstatistical_input,
  idVar = "protein_id",
  intensityStr = "signal_sum_",
  fcStr = "rel_fc_",
  nonZeroCols = "qupm",
  geneNameVar = "gene_name",
  addCol = NULL,
  qualColName = "qupm",
  naStrs = c("NA", "n/d", "NaN"),
  concFactor = 1e6,
  medianNormalizeFC = TRUE,
  filterContaminants = TRUE
)

recomp_sig_df <- recomputeSignalFromRatios(import_df)
preproc_df <- resolveAmbiguousProteinNames(recomp_sig_df)
model_params_df <- getModelParamsDf(
  df = preproc_df
)

fstat_df <- computeFStatFromParams(model_params_df)

set.seed(12, kind = "L'Ecuyer-CMRG")
null_model <- bootstrapNullAlternativeModel(
  df = preproc_df,
  params_df = model_params_df,
  B = 20
)

fdr_tab <- getFDR(
  df_out = fstat_df,
  df_null = null_model
)

hits <- findHits(
  fdr_df = fdr_tab,
  alpha = 0.1
)
TPPFstatistic_hits <- hits %>%
  dplyr::select(clustername, nObs, F_statistic, FDR, slopeH1, detected_effectH1, rssH0, rssH1)

plot2dTppFit(recomp_sig_df, "CHEK2",
  model_type = "H1"
)

save(TPPFstatistic_hits, file = "results/211007_TPPfstathits_UWB24h.RData")
```

Load the results to avoid recalculation

```{r}
load("results/211007_TPPfstathits_UWB24h.RData")

TPPFstatistic_hits_adap <- TPPFstatistic_hits %>%
  mutate(
    mode = ifelse(detected_effectH1 == "stability" & slopeH1 > 0, "stabilized", NA),
    mode = ifelse(detected_effectH1 == "stability" & slopeH1 < 0, "destabilized", mode),
    mode = ifelse(detected_effectH1 != "stability" & slopeH1 < 0, "decreased abundance", mode),
    mode = ifelse(detected_effectH1 != "stability" & slopeH1 > 0, "increased abundance", mode),
    score = sign(slopeH1) * sqrt(rssH0 - rssH1),
    p_adj = FDR,
    pval = NA,
    hit_call = "TPPFstatistic",
    gene_name = clustername,
    representative = mapIds(org.Hs.eg.db, clustername, "UNIPROT", "SYMBOL")
  ) %>%
  dplyr::select(representative, gene_name, hit_call, mode, score, pval, p_adj)
```
```{r number_TPPhits}
ggplot(TPPFstatistic_hits_adap, aes(x= mode, fill=mode)) +
  geom_bar() +
  scale_fill_manual(values = values_effect) +
  cowplot::theme_cowplot() *
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

```
```{r}
ggplot(TPPFstatistic_hits_adap %>% 
         filter(gene_name != "AVD"), 
       aes(x = mode, y = abs(score), colour = mode)) +
  geom_boxplot() +
  scale_colour_manual(values = values_effect) +
  theme_cowplot()
```


## Save

```{r, eval =  FALSE}
TPP_all_scores <- TPPFstatistic_hits_adap
save(TPP_all_scores, file = "results/220530_TPPallscores.RData")
```


# TPP hit evaluation

## Pathways

Check pathway enrichment for de/stabilized hits

```{r}
pathway_check_input <- TPPFstatistic_hits_adap %>%
  filter(p_adj < 0.05)


pathways_universe <- proteins_merged %>%
  separate(gene_name, into = c("representative", "gene_name"), sep = "\\.")%>% 
  distinct(gene_name)

pathway_enrichment <- pathway_check_input %>%
  mutate(enrichment = list(as.data.frame(GSE_RPA(
    geneList = gene_name,
    universe = pathways_universe$gene_name
  ))))

pathway_enrichment_unnest <- pathway_enrichment %>%
  distinct(gene_name, enrichment) %>%
  distinct(enrichment) %>% 
  unnest()

pathway_enrichment_significant <- pathway_enrichment_unnest %>%
  filter(qvalue < 0.05)
```

Plot top 10 pathways for stability score pathway enrichment.

```{r toppathways, fig.width=7, fig.height=4}

# extract top10 pathways for plot
top_TPP_pathways_TPPFstatistic <- pathway_enrichment_significant %>%
  as.data.frame() %>%
  top_n(10, wt = -log10(qvalue))

p_pathway_UWB1.289_TPPFstatistic <- ggplot(top_TPP_pathways_TPPFstatistic, aes(y = -log10(qvalue), x = reorder(Description, -log10(qvalue)))) +
  geom_bar(stat = "identity", position = position_dodge(), fill = "indianred1") +
  geom_hline(yintercept = -log10(0.05)) +
  coord_flip() +
  theme_bw(base_size = 14) +
  theme(axis.text.y = element_text(size = 10)) +
  labs(x = "", title = "Significant pathways\nTPP F-statistic hits")
p_pathway_UWB1.289_TPPFstatistic
```



## Phosphorylation

Assess changes on phosphoproteomic level and their impact on TPP proteins

```{r}
psites <- phospho_ola %>% 
  mutate(gene_name =  GeneSymbol,
                                 aa = str_extract(psite, "[[:upper:]]"),
                                 pos = as.numeric(str_extract(psite, "\\d{1,4}")),
                                 ID_cosmos = paste(GeneSymbol, "_", aa, pos, sep = ""),
                                 ID_cosmos = gsub(",.+$", "", ID_cosmos))%>% 
  dplyr::select(ID_cosmos, gene_name, logFC, t, condition_long, adj.P.Val)

TPP_hits_phosphositelevel <- inner_join(TPPFstatistic_hits_adap, psites, by = "gene_name") %>%
  #filter(hit_call != "footprinting") %>%3
  mutate(significance = ifelse(adj.P.Val < 0.05, "significant", "not significant"))
```

```{r TPP phosphorylation}
volcano_affected_psites <- ggplot(TPP_hits_phosphositelevel,
                                  aes(x = logFC, y = -log10(adj.P.Val))) +
  geom_hline(yintercept = 0, colour = "gray28")+
  geom_vline(xintercept = 0, colour = "gray28")+
  geom_point(aes(colour = significance))+
  geom_text_repel(label = TPP_hits_phosphositelevel$gene_name, size = 3, colour = "gray2") +
  scale_colour_manual(values = c("grey", "darkcyan")) +
  theme_bw(base_size = 14) +
  labs(title = "Phosphorylation on TPP proteins")
volcano_affected_psites
```

# Comparison with footprinting

```{r}
footprints_UWB_ola_hits <- bind_rows("phospho" = filter(viperRes_phospho_corrected, condition_long == "UWB1.289_24h_4uM" ) %>% top_n(n = 30, wt = absNES),
                                      "trans" =filter(viperRes_trans, condition_long == "UWB1.289_24h_4uM" ) %>% top_n(n = 30, wt = absNES), 
                                     .id = "omic") %>% 
  mutate(mode = ifelse(NES > 0, "active", "inactive"), hit_call = "footprinting") %>% 
  dplyr::select(gene_name = regulator, hit_call, mode, score = NES, type = omic)

hit_comparisons <- bind_rows(footprints_UWB_ola_hits, TPPFstatistic_hits_adap) %>%
  mutate(type = ifelse(is.na(type), "TPP", type)) %>%
  dplyr::select(-representative) %>%
  group_by(gene_name) %>%
  mutate(n_scores = n())%>%
  group_by(hit_call) %>%
  mutate(n_proteins = n()) %>% 
  ungroup()%>%   
  filter(hit_call %in% c("footprinting", "TPPFstatistic")) %>% 
  mutate(hit_call = gsub("TPPFstatistic", "TPP", hit_call), 
        hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% dorothea$source_genesymbol), "TF", hit_call),
         hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% KSN$enzyme_genesymbol), "Kinase", hit_call),
        mode = ifelse(mode == "increased abundance", "stabilized", mode),
        mode = ifelse(mode == "decreased abundance", "destabilized", mode),
        mode = factor(mode, levels = c("stabilized","destabilized",  "active", "inactive")))
```

## Overlaps

```{r fig.width=4.5, fig.height=2}
values_paper <- c("Kinase" = "darkcyan", "TF" = "maroon4", "TPP" = "mediumseagreen")
protein_order_small <- hit_comparisons %>%
  dplyr::group_by(gene_name) %>% 
  dplyr::mutate(n_scores = n()) %>% 
  ungroup() %>%
  dplyr::select(gene_name, n_scores) %>%
  distinct(gene_name, .keep_all = T) %>%
  arrange(-n_scores)

p_comp <- hit_comparisons %>% 
  ggplot(aes(y = hit_call,
             x = factor(gene_name, levels = protein_order_small$gene_name))) +
  geom_raster(aes(fill = hit_call, alpha = mode), 
              colour = "white") +
  scale_fill_manual(values = values_paper, name = "") +
  scale_alpha_manual(values = c(1, 0.7, 1, 0.7), guide = F) +
  #theme_bw(base_size = 18) +
  cowplot::theme_cowplot()+
  theme(axis.text.x = element_blank(),  axis.ticks.x=element_blank(),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.position = "right")+
  labs(x = "all proteins", y = "")
p_comp
```

### DDR

```{r}
library(reactome.db)
pathwaymembers <- toTable(reactomePATHID2EXTID) %>% 
  mutate(gene_name = mapIds(org.Hs.eg.db, gene_id, 'SYMBOL', 'ENTREZID'))

pathway_names <-  toTable(reactomePATHID2NAME) %>% 
  filter(grepl("Homo sapiens", path_name))

DDR <- pathway_names %>% 
  filter(grepl("DNA Damage|DNA damage|DNA Repair|DNA repair|Checkpoint|Repair|Damage|repair|damage| Strand Breaks|DNA double-strand break|Homologous Recombination|Chk1/Chk2|DNA Double Strand Break|cell cycle|Cell Cycle|NIMA|APC|G2/|G1/|Mitotic|mitotic|Prometaphase|G2 Phase|G1 Phase|Prophase|cyclin|Cyclin", path_name))

DDR_members <- pathwaymembers %>% 
  filter(DB_ID %in% DDR$DB_ID) %>% 
  distinct(gene_name)

functional_coverage_TPP <- inner_join(DDR_members,hit_comparisons,  by = c("gene_name"))%>% 
  mutate(hit_call = gsub("TPPFstatistic", "TPP", hit_call)) %>%
  group_by(gene_name) %>%
  mutate(n_scores = n()) 
```


```{r}
functional_coverage_TPP <- inner_join(DDR_members,hit_comparisons,  by = c("gene_name"))%>% 
  mutate(hit_call = gsub("TPPFstatistic", "TPP", hit_call), 
          hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% dorothea$source_genesymbol), "TF", hit_call),
          hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% KSN$enzyme_genesymbol), "Kinase", hit_call),
         mode = factor(mode, levels = c("stabilized","destabilized",  "active", "inactive"))) %>% 
  group_by(gene_name) %>%
  mutate(n_scores = n()) 


protein_order <-  functional_coverage_TPP %>%
  dplyr::group_by(gene_name) %>% 
  dplyr::mutate(n_scores = n()) %>% 
  arrange(n_scores, hit_call) %>% 
    distinct(gene_name)

p_DNArepair <- functional_coverage_TPP%>% 
  ggplot(aes(x = factor(gene_name, levels = protein_order$gene_name),
             y = hit_call)) +
  geom_raster(aes(fill=hit_call, alpha = mode))+
  scale_fill_manual(values =values_paper) +
  scale_alpha_manual(values = c(1, 0.7, 1, 0.7), guide = F) +
  cowplot::theme_cowplot() +
    theme(axis.text.x = element_blank(),axis.ticks.x =  element_blank(), legend.position = "right") +
  labs(x = "proteins of DDR",
       y= "") +
  guides(fill = guide_legend(title = ""))
p_DNArepair
```

## Phospho

```{r}
TPP_hits_phosphositelevel <- inner_join(hit_comparisons, psites, by = "gene_name") %>%
  mutate(significance = ifelse(adj.P.Val < 0.05, "significant", "not significant"))
```


```{r}
my_comparisons <- list( c("Kinase", "TPP"), c("TF", "TPP"))
p <- TPP_hits_phosphositelevel %>% 
   mutate(hit_call = gsub("TPPFstatistic", "TPP", hit_call), 
          hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% dorothea$source_genesymbol), "TF", hit_call),
          hit_call = ifelse(hit_call == "footprinting" & (gene_name %in% KSN$enzyme_genesymbol), "Kinase", hit_call)) %>% 
  ggplot(aes(hit_call, logFC)) +
  geom_boxplot(aes(colour = hit_call), lwd =0.8) +
  ggpubr::stat_compare_means(comparisons = my_comparisons, label.y = c(1.5, 1.3))+
  scale_colour_manual(values = values_paper, guide = "none") +
  cowplot::theme_cowplot() +
  labs(x = "", y = "log2 fold-change \n phosphosite intensity")
p

```

### CDK1 example


```{r}
kinases_downstream <- filter(KSN, enzyme_genesymbol %in% footprints_UWB_ola_hits$gene_name) %>%
  mutate(substrates = str_extract(substrate_genesymbol, "^.*?(?=_)")) %>%
  dplyr::select(enzyme_genesymbol, substrates, sign) %>%
  distinct(enzyme_genesymbol, substrates, sign)

TPP_kinase_downstream_overlap <- inner_join(hit_comparisons, kinases_downstream, by = c("gene_name" = "substrates")) %>%
  group_by(hit_call, enzyme_genesymbol) %>%
  mutate(n_enz = n()) %>%
  group_by(hit_call, gene_name) %>%
  mutate(n_TPP = n()) %>% 
  filter(hit_call == "TPP")
```


```{r}
kinase_example <- TPP_kinase_downstream_overlap %>% 
  filter(enzyme_genesymbol == "CDK1") %>% 
  ungroup() %>% 
  dplyr::select(from = enzyme_genesymbol, to = gene_name, sign) %>% 
  mutate(dashes = ifelse(sign == 1, FALSE, TRUE))%>% filter(to %in% c("WWTR1", "CDCA5", "CKAP2", "KIF11"))


example_phospho <- TPP_hits_phosphositelevel %>% 
  filter(gene_name %in% kinase_example$to) %>% 
  dplyr::select(from = gene_name, to = ID_cosmos) %>% 
  mutate(sign = 1, dashes = FALSE)

kinase_example <- bind_rows(kinase_example,example_phospho) %>% 
  mutate(color = "grey", length = ifelse(grepl("_", to), 10, 300))

nodes <- data.frame(id = unique(c(kinase_example$from, kinase_example$to))) %>% 
  mutate(group = ifelse(id == "CDK1", "Kinase", "TPP"),
         group = ifelse(grepl("_", id), "phosphosite", group),
         shape = ifelse(group == "phosphosite", "dot", "square"),
         label = gsub("^.*_", "",id), 
         level = ifelse(group == "kinase", 1, 2),
         level = ifelse(group == "phosphosite", 3, level),
         level = ifelse(id == "CDK1", 1, level))

p_kinase_example = visNetwork(nodes,kinase_example, height = "800px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "phosphosite", color = "darkseagreen") %>%
    visEdges(physics = F, width = 2) %>% 
  visNodes(font = '30px Arial black', physics = F) %>% 
  visHierarchicalLayout(levelSeparation = 300, nodeSpacing = 70)
    # visLegend() %>%
    # visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
  #visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_kinase_example
```


# TPP activities

To estimate the activity of TPP hits we searched for consistent edges between upstream kinases identified by viper and downstream TPP proteins.

```{r}
# load viper results
viper_phospho_top <- viperRes_phospho_corrected %>% 
  group_by(condition_long) %>% 
  top_n(30, wt = absNES)

viper_trans_top <- viperRes_trans %>% 
  group_by(condition_long) %>% 
  top_n(30, wt = absNES)
```


```{r}
# filter prior knowledge links
annoated_ptms_measured <- KSN %>%
  separate(substrate_genesymbol, into = c("substrate_genesymbol", "site"), sep = "_") %>% 
  # subset prior knowledge to subset with measured kinase and target
  filter(enzyme_genesymbol %in% viper_phospho_top$regulator &
    substrate_genesymbol %in% TPPFstatistic_hits_adap$gene_name) %>%
  left_join(viper_phospho_top, by = c("enzyme_genesymbol" = "regulator")) %>%
  # predict for each psite an activation state based on the upstream kinase activity
  mutate(Kinase_activity = sign(NES)) %>%
  dplyr::select(enzyme_genesymbol, sign, substrate_genesymbol, site, Kinase_activity) %>%
  mutate(substrate_activity_predicted =sign*Kinase_activity)
```


## Combine TPP activity information for COSMOS

```{r}
TPP_cosmos_input <- annoated_ptms_measured %>% 
  # only keep TPP proteins for which there is consistent activation or inactivation information 
  # necessary because we do not no which upstream kinase is potentially regulating the TPP protein
  # the others can be modelled active or inactive supported by (de)stabilization information
  distinct(TPP_protein = substrate_genesymbol, substrate_activity_predicted) %>% 
  group_by(TPP_protein) %>%
  summarise(substrate_activity_predicted_summ = sign(mean(substrate_activity_predicted)))

# aggregate final COSMOS input df
TPP_cosmos_input_final <- TPP_cosmos_input %>%
  mutate(substrate_activity_predicted_summ = ifelse(substrate_activity_predicted_summ == 0, NaN, substrate_activity_predicted_summ)) %>%
  bind_rows(data.frame(TPP_protein = TPPFstatistic_hits_adap$gene_name, substrate_activity_predicted_summ = NaN)) %>%
  distinct(TPP_protein, .keep_all = T)
```

# Remove inconsistent edges

To support the optimization to model consistent links between kinases and TPP proteins as assumed for the activity estimation, we removed incosistent prior knowledge edges. 

For example: Kinase (active) --activates_via_S10--> TPP protein (S10 dephosphorylated) would be removed, because the dephosphorylation of this psite can not be explained using this link (but this information is not always reflected in TPP activity).

To do this correction we use the abundance corrected version of the phosphoproteomics data because we are interested in exclusively phosphorylation driven changes.

```{r}
# Format phosphoproteomics data after intensity correction
psites_corrected <- limma_UWB1.289_initialdataset_correctedphospho %>%
  filter(condition == "BRCA_nul_24h_4" & comparison == "Ola") %>%
  mutate(
    gene_name = mapIds(org.Hs.eg.db, representative, "SYMBOL", "UNIPROT"),
    condition_long = paste(condition, comparison, sep = "_"),
    UP = representative,
    GeneSymbol = gene_name,
    aa = str_extract(psite, "[[:upper:]]"),
    pos = as.numeric(str_extract(psite, "\\d{1,4}"))
  ) %>%
  mutate(ID_cosmos = paste(GeneSymbol, "_", aa, pos, sep = "")) %>%
  dplyr::select(ID_cosmos, logFC, t, condition_long) %>%
  filter(!is.na(logFC) & !is.na(ID_cosmos)) %>%
  mutate(ID_cosmos = gsub(",.+$", "", ID_cosmos))


# format KSN

KSN <- KSN %>% 
  separate(col = substrate_genesymbol, c("substrate_protein", "psite"), remove = FALSE)
```

Format footprinting and TPP results and prepare for end object which is exported for COSMOS.

```{r}
Kinases_input_COSMOS <- sign(viper_phospho_top$NES)
names(Kinases_input_COSMOS) <- paste0(
  "X",
  mapIds(org.Hs.eg.db, viper_phospho_top$regulator, "ENTREZID", "SYMBOL")
)
Kinases_input_COSMOS <- Kinases_input_COSMOS[names(Kinases_input_COSMOS) != "XNA"]

TPPFstatistic_hits_adap <- TPPFstatistic_hits_adap %>% 
  mutate(signed_score = ifelse(mode == "destabilized" | mode == "decreased abundance", -1, 1))
```

## Combine prior knowledge, phosphorylation and TPP

Perform the kinase sign filtering to remove links btw kianses and TPP proteins

```{r}
# extract kinase targets from TPP proteins
kinase_targets <- TPPFstatistic_hits_adap %>% 
  filter(gene_name %in% KSN$substrate_protein) %>% 
  mutate(cosmosID = paste0("X", mapIds(org.Hs.eg.db,gene_name, "ENTREZID", "SYMBOL")))

# merge with limma results
KSN_withFCs <- KSN %>%
  left_join(psites, by = c("substrate_genesymbol" = "ID_cosmos")) %>%
  dplyr::select(-condition_long)
KSN_TPPtargets <- KSN_withFCs %>%
  filter(substrate_protein %in% kinase_targets$gene_name) %>% 
  mutate(source_cosmos =  paste0("X", mapIds(org.Hs.eg.db, enzyme_genesymbol, "ENTREZID", "SYMBOL")))

# use edges with viper kinase
inputs <- Kinases_input_COSMOS[names(Kinases_input_COSMOS) %in% KSN_TPPtargets$source_cosmos]

# merge
KSN_TPPtargets$kinase_sign <- sapply(KSN_TPPtargets$source_cosmos, function(x, inputs) {
  if (x %in% names(inputs)) {
    return(as.numeric(inputs[x]))
  } else {
    return(NA)
  }
}, inputs = inputs)

# remove edges without kinase measurement
KSN_TPPtargets_filt <- filter(KSN_TPPtargets, !is.na(logFC) & !is.na(kinase_sign))
# get inconsistent edges
KSN_TPPtargets_filt <- KSN_TPPtargets_filt %>% 
  filter(sign(logFC * sign) != sign(kinase_sign)) %>% 
  select(substrate_genesymbol, enzyme_genesymbol, sign, logFC, kinase_sign)

KSN_TPPtargets_filt
```


# Save

```{r, eval = F}
save(TPP_cosmos_input_final, viper_trans_top, viper_phospho_top, KSN_TPPtargets_filt,
    file = "results/220512_cosmosinput_merged.RData")
```

