---
title: "Script to run adjusted COSMOS workflow with TPP data"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide
---


# General settings


```{r setup}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  include = TRUE,
  cache = TRUE,
  cache.lazy = FALSE,
  eval = TRUE,
  fig.width = 8 * (1 + sqrt(5)) / 2, fig.height = 8 # golden ratio
)

knitr::opts_knit$set(root.dir = "C:/Users/burtsche/Documents/COSMOS-TPP_paper")
```

## Load packages

```{r, message=F, warning =F}
library(plyr)
library(tidyverse) # ggplot2, purrr, tibble, dplyr, tidyr, stringr, readr, forcats
library("reshape2")
library(OmnipathR)
library(viridis)
library(ggrepel)
# options(connectionObserver = NULL)
library(org.Hs.eg.db)
library(visNetwork)
library(knitr)
library(ggpubr)
library(cosmosR)
library(CARNIVAL)
library(igraph)
library(biomaRt)

mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
#devtools::install_github("saezlab/CARNIVAL@v1.3")
```

## Run settings

Set a run ID which will be used to save the results.

```{r}
source("src/revision_COSMOS_functions.R")
runID <- paste(gsub(" |-|:", "_", Sys.time()), "testforpaper_onThinkPAd_V1", sep = "_")
```

## Load data

```{r}
load("data/210626_DEmultiomics_Olaparib_gathered_updated")
#load("data/2021_09_10_TPPscores_Fstatmono_T47D_without30uM.RData")
load("data/310521_PKNcollection.RData")
load("results/220512_cosmosinput_merged.RData")
load("results/220530_TPPallscores.RData")

```

## Define TPP input

```{r}
TPP_cosmos_input_final <- TPP_cosmos_input_final# input object in correct format for TPP input
TPP_all_scores <- TPP_all_scores # all scores to extarct stabilized and destabilized weights

# top_mono <- TPP_all_scores %>%
#   filter(hit_call == "TPPmonotony") %>%
#   top_n(40, wt =abs(score))

# TPP_cosmos_input_final <- TPP_cosmos_input_final_monotony %>% filter(TPP_protein %in% top_mono$gene_name)

TPP_all_scores <- TPP_all_scores %>%
  mutate(signed_score = ifelse(mode == "destabilized" | mode == "decreased abundance", -1, 1)) %>%
  mutate(weight = ifelse(hit_call == "TPPFstatistic" | hit_call == "TPPmonotony", score, signed_score)) %>%
  filter(hit_call == "TPPFstatistic")
```

#Prepare input data

##Footprinting regulators


Code for new footprinting results. They are already filtered in the COSMOSinput object after kinase sign filtering script.

```{r}
TF_input_COSMOS <- sign(viper_trans_top$NES)
names(TF_input_COSMOS) <- paste0(
  "X",
  mapIds(org.Hs.eg.db, viper_trans_top$regulator, "ENTREZID", "SYMBOL")
)


Kinases_input_COSMOS <- sign(viper_phospho_top$NES)
names(Kinases_input_COSMOS) <- paste0(
  "X",
  mapIds(org.Hs.eg.db, viper_phospho_top$regulator, "ENTREZID", "SYMBOL")
)
```

##TPP hits

```{r}
TPP_all_scores_v <- TPP_all_scores$weight
names(TPP_all_scores_v) <- paste0(
  "X",
  mapIds(org.Hs.eg.db, TPP_all_scores$gene_name, "ENTREZID", "SYMBOL")
)
TPP_all_scores_v <- TPP_all_scores_v[names(TPP_all_scores_v) != "XNA"]

TPP_input_COSMOS_tmp <- as.numeric(TPP_cosmos_input_final$substrate_activity_predicted_summ)
names(TPP_input_COSMOS_tmp) <- paste0(
  "X",
  mapIds(org.Hs.eg.db, TPP_cosmos_input_final$TPP_protein, "ENTREZID", "SYMBOL")
)
TPP_input_COSMOS_tmp <- TPP_input_COSMOS_tmp[names(TPP_input_COSMOS_tmp) != "XNA"]
```

Weights for TPP hits without activity estimate

```{r}
TPP_input_COSMOS_weights_tmp <- TPP_input_COSMOS_tmp[is.na(TPP_input_COSMOS_tmp)]
names(TPP_input_COSMOS_weights_tmp) <- names(TPP_input_COSMOS_weights_tmp)
TPP_input_COSMOS_weights <- TPP_all_scores_v[names(TPP_input_COSMOS_weights_tmp)]

rm(TPP_input_COSMOS_weights_tmp)

```


##RNA-Seq data

Extract transcriptomic background for optimization.

```{r}
transcriptomics_filtered <- transcriptomics_ola %>%
  filter(condition_long == "UWB1.289_24h_4uM" &
    !is.na(EntrezID)) %>%
  distinct(EntrezID, .keep_all = T)

RNA_input_COSMOS <- transcriptomics_filtered$logFC
names(RNA_input_COSMOS) <- as.character(paste0("X", transcriptomics_filtered$EntrezID))
RNA_input_COSMOS <- RNA_input_COSMOS[names(RNA_input_COSMOS) != "XNA"]
names(RNA_input_COSMOS) <- as.character(names(RNA_input_COSMOS))

transcriptomics_signed <- transcriptomics_filtered %>%
  dplyr::select(EntrezID, GeneSymbol, logFC) %>%
  mutate(direction = sign(logFC))
transcriptomics_filtered$ID <- paste0("X", transcriptomics_filtered$EntrezID)
```

Create data.frame which is saved in result object

```{r}
expression_data <- transcriptomics_filtered$FC %>% as.vector()
names(expression_data) <- transcriptomics_filtered$ID
```

##Proteomics

Extract proteomic background for optimization.

```{r}
proteomics_filtered <- proteomics_ola %>%
  filter(condition_long == "UWB1.289_24h_4uM" &
    !is.na(EntrezID)) %>%
  distinct(EntrezID, .keep_all = T)

proteomics_signed <- proteomics_filtered %>%
  dplyr::select(EntrezID, GeneSymbol, logFC) %>%
  mutate(direction = sign(logFC))
```

###RNA vs Protein

Compare RNA and protein signal to use this later in PKN filtering. As we have expression proteomics data, we correct some assumptions made based on transcriptomics data. Proteins/genes for which the trans and prot sign do not agree are corrected.

```{r}
trans_vs_prot <- transcriptomics_signed %>%
  inner_join(proteomics_signed, by = "GeneSymbol") %>%
  mutate(
    rank_trans = dense_rank(logFC.x),
    rank_prot = dense_rank(logFC.y)
  )
trans_vs_prot_direction <- trans_vs_prot %>%
  mutate(agreement = ifelse(direction.x == direction.y, "same direction", "other direction"))

p_trans_vs_prot_direction <- ggplot(filter(trans_vs_prot_direction, !is.na(agreement)), aes(x = agreement)) +
  geom_bar()
p_trans_vs_prot_direction
```

Prepare final data frame to filter metaPKN later.

```{r}
transcriptomics_proteomics_filtered <- transcriptomics_filtered %>%
  left_join(dplyr::select(trans_vs_prot_direction, GeneSymbol, agreement), by = "GeneSymbol") %>%
  mutate(agreement = ifelse(is.na(agreement), "no overlap", agreement)) %>%
  mutate(FC = ifelse(agreement == "other direction", 0, FC))
```


##PKN

### Choose PKN

Define which PKN to use

```{r}
# meta_network <- GSK_PKN %>%
#   filter(!is.na(source) & !is.na(target)) %>%
#   mutate(source = as.character(paste0("X", mapIds(org.Hs.eg.db, source, 'ENTREZID', 'SYMBOL'))),
#   target = as.character(paste0("X", mapIds(org.Hs.eg.db, target, 'ENTREZID', 'SYMBOL')))) %>%
#   filter(source != "XNA" & target != "XNA") %>%
#   setNames(c("source", "interaction", "target"))

meta_network <- COSMOS_PKN_new %>%
  filter(!is.na(source) & !is.na(target)) %>%
  setNames(c("source", "interaction", "target"))
```

### Kinase sign filtering

Filter inconsistent edges btw TPP and Kinases (From Kinase sign filtering)

```{r}
KSN_TPPtargets_filt <- KSN_TPPtargets_filt %>%
  separate(substrate_genesymbol, into = c("substrate_protein","substrate_site"), remove = F) %>% 
  mutate(
    source = as.character(paste0("X", mapIds(org.Hs.eg.db, enzyme_genesymbol, "ENTREZID", "SYMBOL"))),
    target = as.character(paste0("X", mapIds(org.Hs.eg.db, substrate_protein, "ENTREZID", "SYMBOL")))
  ) %>%
  filter(source != "XNA" & target != "XNA")

meta_network <- meta_network %>%
  anti_join(KSN_TPPtargets_filt, by = c("source", "target"))
```

## Set signaling input

```{r}
signaling_input_COSMOS <- c(Kinases_input_COSMOS, TPP_input_COSMOS_tmp)

# remove overlpas of TPP and Kinases, keep kinase activity estimate
signaling_input_COSMOS <- signaling_input_COSMOS[!duplicated(names(signaling_input_COSMOS))]

# sbring in CARNIVAL accepted format
names(signaling_input_COSMOS) <- as.character(names(signaling_input_COSMOS))
signaling_input_COSMOS <- t(as.matrix(signaling_input_COSMOS)) %>% as.data.frame()
```


# Forward runs

## Set TF input

```{r}
TF_input_COSMOS <- t(c(TF_input_COSMOS)) %>% as.data.frame()
```

## Set weights

```{r}
weights_COSMOS <- t(TPP_input_COSMOS_weights) %>% as.data.frame()
weights_COSMOS <- weights_COSMOS[!is.na(names(weights_COSMOS))]

# scale weights btw 1 and -1 --> does not change results in comp to discrete 1/-1 weights
range_COSMOSweights <- function(x) {
  2 * ((x - min(x)) / (max(x) - min(x))) - 1
}

weights_COSMOS <- range_COSMOSweights(weights_COSMOS)
```

<!-- ## Expand PKN for TPP hits -->

<!-- ```{r} -->
<!-- TPP_interactions <- filter(meta_network, source %in% names(TPP_input_COSMOS) | source %in% names(TPP_input_COSMOS)) -->
<!-- TPP_interactions_reverted <- mutate(TPP_interactions, interaction = interaction * (-1) ) -->
<!-- expanded_meta_network <- rbind(meta_network, TPP_interactions_reverted) -->
<!-- ``` -->


## Filter PKN: expressed genes

Filter metaPKN for nodes relevant acording to expressed gene list (from RNA seq)

```{r}
# meta_network_expressionfiltered <- expanded_meta_network

meta_network_expressionfiltered <- meta_network

signaling <- colnames(signaling_input_COSMOS)
measurement <- colnames(TF_input_COSMOS)

expressed_gene_list <- c(names(RNA_input_COSMOS), signaling, measurement)
expressed_gene_list <- gsub("X", "", expressed_gene_list) %>% unique()


meta_network_expressionfiltered$source <-
  sapply(meta_network_expressionfiltered$source, is_expressed)
meta_network_expressionfiltered <-
  meta_network_expressionfiltered[complete.cases(meta_network_expressionfiltered), ]
meta_network_expressionfiltered$target <-
  sapply(meta_network_expressionfiltered$target, is_expressed)
meta_network_expressionfiltered <-
  meta_network_expressionfiltered[complete.cases(meta_network_expressionfiltered), ]
```

## Filter input for PKN occurence

```{r, message=FALSE, warning=FALSE}
# filter inputs for thier occurance in the network
TF_input_COSMOS_inPKN <- filter_inputs(TF_input_COSMOS, meta_network_expressionfiltered)

signaling_input_COSMOS_inPKN <- filter_inputs(signaling_input_COSMOS, meta_network_expressionfiltered)

# filter overlaps
signaling_input_COSMOS_inPKN <- signaling_input_COSMOS_inPKN[, !(colnames(signaling_input_COSMOS_inPKN) %in% colnames(TF_input_COSMOS_inPKN))]
```


# Fw1

## Filter input for fw1

```{r, message=FALSE, warning=FALSE}
# filter network for nodes which are maximum 8 nodes from measurement
meta_network_final_fw1 <- downstream_neighbours(
  meta_network_expressionfiltered,
  8,
  names(TF_input_COSMOS_inPKN)
)

# filter network for correct TF interaction signs
# dorothea <- format_dorothea()

load("data/dorothea.RData")

meta_network_final_fw1_t1 <- filter_TF_sign(
  meta_network = meta_network_final_fw1,
  ttop_RNA = transcriptomics_filtered,
  inputs = TF_input_COSMOS_inPKN,
  TF_targets = dorothea
)


meta_network_final_fw1_t2 <- filter_TF_sign(
  meta_network = meta_network_final_fw1,
  ttop_RNA = transcriptomics_proteomics_filtered,
  inputs = TF_input_COSMOS_inPKN,
  TF_targets = dorothea
)
```


<!-- Edges which are filtered because of proteomics and transcriptomics discrepancies. -->

<!-- ```{r} -->
<!-- filtered_edges <- anti_join(meta_network_final_fw1_t1, meta_network_final_fw1_t2) -->
<!-- ``` -->

```{r, message=FALSE, warning=FALSE}
meta_network_final_fw1 <- meta_network_final_fw1_t2

# filter inputs for thier occurance in the network
measurements_fw1 <- filter_inputs(TF_input_COSMOS_inPKN, meta_network_final_fw1)
input_fw1 <- filter_inputs(signaling_input_COSMOS_inPKN, meta_network_final_fw1)
```


## Run CARNIVAL

```{r, eval = FALSE}
# define options for CARNIVAL optimization
timelimit <- 800
mipGAP <- 0.15
solverPath <- "C:/Program Files/IBM/ILOG/CPLEX_Studio1210/cplex/bin/x64_win64/cplex.exe"
used_options <- c(as.character(timelimit), as.character(mipGAP), solverPath)

#beepr::beep(1)
CARNIVAL_Result_fw1 <- runCARNIVAL(
  solverPath = solverPath,
  netObj = meta_network_final_fw1,
  measObj = measurements_fw1,
  inputObj = input_fw1,
  weightObj = weights_COSMOS,
  timelimit = timelimit,
  mipGAP = mipGAP,
  solver = "cplex"
) # ,
# dir_name = "/Users/miraburtscher/Documents/gsk/O2_Network_modeling/tmp")

COSMOS_object_fw1 <- structure(list(
  meta_network = meta_network_final_fw1,
  tf_regulon = dorothea,
  signaling_data_UP = input_fw1,
  metabolic_data_DOWN = measurements_fw1,
  expression_data = expression_data,
  weightedSIF = as.data.frame(CARNIVAL_Result_fw1$weightedSIF),
  nodeAtt = as.data.frame(CARNIVAL_Result_fw1$nodesAttributes),
  history = c(runID)
),
used_options = used_options,
class = "cosmos_data"
)

#beepr::beep(3)
```

# Fw2

## Filter input for fw2

```{r, message=FALSE, warning=FALSE, eval = FALSE}
TF_signs_fw1 <- get_TF_sign_from_CARNI(CARNIVAL_Result_fw1, dorothea)
meta_network_final_fw2_t1 <- filter_TF_sign(
  meta_network = meta_network_final_fw1,
  ttop_RNA = transcriptomics_filtered,
  inputs = TF_signs_fw1,
  TF_targets = dorothea
)

meta_network_final_fw2_t2 <- filter_TF_sign(
  meta_network = meta_network_final_fw1,
  ttop_RNA = transcriptomics_proteomics_filtered,
  inputs = TF_signs_fw1,
  TF_targets = dorothea
)

meta_network_final_fw2 <- meta_network_final_fw2_t2

measurements_fw2 <- filter_inputs(TF_input_COSMOS_inPKN, meta_network_final_fw2)
input_fw2 <- filter_inputs(signaling_input_COSMOS_inPKN, meta_network_final_fw2)
```

## Run CARNIVAL

```{r, eval = FALSE}
# define options for CARNIVAL optimization
timelimit <- 500
mipGAP <- 0.15
#solverPath <- "~/Documents/gsk/COSMOS/CPLEX_Studio1210/cplex/bin/x86-64_osx/cplex"
used_options <- c(as.character(timelimit), as.character(mipGAP), solverPath)

beepr::beep(1)
CARNIVAL_Result_fw2 <- runCARNIVAL(
  solverPath = solverPath,
  netObj = meta_network_final_fw2,
  measObj = measurements_fw2,
  inputObj = input_fw2,
  weightObj = weights_COSMOS,
  timelimit = timelimit,
  mipGAP = mipGAP,
  solver = "cplex"
) # ,
# dir_name = "/Users/miraburtscher/Documents/gsk/O2_Network_modeling/tmp")

COSMOS_object_fw2 <- structure(list(
  meta_network = meta_network_final_fw2,
  tf_regulon = dorothea,
  signaling_data_UP = input_fw2,
  metabolic_data_DOWN = measurements_fw2,
  expression_data = expression_data,
  weightedSIF = as.data.frame(CARNIVAL_Result_fw2$weightedSIF),
  nodeAtt = as.data.frame(CARNIVAL_Result_fw2$nodesAttributes),
  history = c(runID)
),
used_options = used_options,
class = "cosmos_data"
)
beepr::beep(3)
```

# Reverse runs

Set up new run hierarchy

## TF input

```{r}
TF_input_COSMOS <- sign(viper_trans_top$NES)

names(TF_input_COSMOS) <- paste0("X", mapIds(org.Hs.eg.db, viper_trans_top$regulator, "ENTREZID", "SYMBOL"))

TF_input_COSMOS <- TF_input_COSMOS[names(TF_input_COSMOS) != "XNA"]
names(TF_input_COSMOS) <- as.character(names(TF_input_COSMOS))

TF_input_COSMOS <- t(c(TF_input_COSMOS, TPP_input_COSMOS_tmp)) %>% as.data.frame()
TF_input_COSMOS <- TF_input_COSMOS[!duplicated(names(TF_input_COSMOS))]
```

## Signaling input

```{r}
signaling_input_COSMOS <- t(Kinases_input_COSMOS) %>% as.data.frame()
```

## Weights

```{r}
weights_COSMOS <- t(TPP_input_COSMOS_weights) %>% as.data.frame()
weights_COSMOS <- weights_COSMOS[!is.na(names(weights_COSMOS))]

# scale weights
weights_COSMOS <- range_COSMOSweights(weights_COSMOS)
```

## Filter PKN: expressed genes

Filter metaPKN for nodes relevant acording to expressed gene list

```{r}
# meta_network_expressionfiltered <- expanded_meta_network
meta_network_expressionfiltered <- meta_network

signaling <- colnames(signaling_input_COSMOS)
measurement <- colnames(TF_input_COSMOS)

expressed_gene_list <- c(names(RNA_input_COSMOS), signaling, measurement)
expressed_gene_list <- gsub("X", "", expressed_gene_list) %>% unique()


meta_network_expressionfiltered$source <-
  sapply(meta_network_expressionfiltered$source, is_expressed)
meta_network_expressionfiltered <-
  meta_network_expressionfiltered[complete.cases(meta_network_expressionfiltered), ]
meta_network_expressionfiltered$target <-
  sapply(meta_network_expressionfiltered$target, is_expressed)
meta_network_expressionfiltered <-
  meta_network_expressionfiltered[complete.cases(meta_network_expressionfiltered), ]
```

## Filter input for PKN occurence

```{r, message=FALSE, warning=FALSE}
# filter inputs for thier occurance in the network
TF_input_COSMOS_inPKN <- filter_inputs(TF_input_COSMOS, meta_network_expressionfiltered)
signaling_input_COSMOS_inPKN <- filter_inputs(signaling_input_COSMOS, meta_network_expressionfiltered)

# filter overlaps
signaling_input_COSMOS_inPKN <- signaling_input_COSMOS_inPKN[, !(colnames(signaling_input_COSMOS_inPKN) %in% colnames(TF_input_COSMOS_inPKN))]

# filter one node because of strange error message
signaling_input_COSMOS_inPKN <- signaling_input_COSMOS_inPKN[!grepl("X1786$", names(signaling_input_COSMOS_inPKN))]
names(signaling_input_COSMOS_inPKN) <- gsub("\\.1", "", names(signaling_input_COSMOS_inPKN))
```

# Rev1

## Filter input for Rev1

```{r, message=FALSE, warning=FALSE}
# filter network for nodes which are maximum 8 nodes from measurement
meta_network_final_rev1 <- downstream_neighbours(
  meta_network_expressionfiltered,
  8,
  names(signaling_input_COSMOS_inPKN)
)

# filter network for correct TF interaction signs
# dorothea <- format_dorothea()
meta_network_final_rev1_t1 <- filter_TF_sign(
  meta_network = meta_network_final_rev1,
  ttop_RNA = transcriptomics_filtered,
  inputs = TF_input_COSMOS_inPKN,
  TF_targets = dorothea
)

meta_network_final_rev1_t2 <- filter_TF_sign(
  meta_network = meta_network_final_rev1,
  ttop_RNA = transcriptomics_proteomics_filtered,
  inputs = TF_input_COSMOS_inPKN,
  TF_targets = dorothea
)

meta_network_final_rev1 <- meta_network_final_rev1_t2

# filter inputs for thier occurance in the network
measurements_rev1 <- filter_inputs(TF_input_COSMOS_inPKN, meta_network_final_rev1)
input_rev1 <- filter_inputs(signaling_input_COSMOS_inPKN, meta_network_final_rev1)
```

## Run CARNIVAL

```{r, eval = FALSE}

# define options for CARNIVAL optimization
timelimit <- 500
mipGAP <- 0.15

# solverPath <- "~/Documents/gsk/COSMOS/CPLEX_Studio1210/cplex/bin/x86-64_osx/cplex"
used_options <- c(as.character(timelimit), as.character(mipGAP), solverPath)

beepr::beep(1)
CARNIVAL_Result_rev1 <- runCARNIVAL(
  solverPath = solverPath,
  netObj = meta_network_final_rev1,
  measObj = measurements_rev1,
  inputObj = input_rev1,
  weightObj = weights_COSMOS,
  timelimit = timelimit,
  mipGAP = mipGAP,
  solver = "cplex",
  dir_name = "/Users/miraburtscher/Documents/gsk/O2_Network_modeling/tmp"
)

COSMOS_object_rev1 <- structure(list(
  meta_network = meta_network_final_rev1,
  tf_regulon = dorothea,
  signaling_data_UP = input_rev1,
  metabolic_data_DOWN = measurements_rev1,
  expression_data = expression_data,
  weightedSIF = as.data.frame(CARNIVAL_Result_rev1$weightedSIF),
  nodeAtt = as.data.frame(CARNIVAL_Result_rev1$nodesAttributes),
  history = c(runID)
),
used_options = used_options,
class = "cosmos_data"
)
beepr::beep(3)
```

# rev2

## Filter input for rev2

```{r, message=FALSE, warning=FALSE, eval = FALSE}
TF_signs_rev1 <- get_TF_sign_from_CARNI(CARNIVAL_Result_rev1, dorothea)
meta_network_final_rev2_t1 <- filter_TF_sign(
  meta_network = meta_network_final_rev1,
  ttop_RNA = transcriptomics_filtered,
  inputs = TF_signs_rev1,
  TF_targets = dorothea
)

meta_network_final_rev2_t2 <- filter_TF_sign(
  meta_network = meta_network_final_rev1,
  ttop_RNA = transcriptomics_proteomics_filtered,
  inputs = TF_signs_rev1,
  TF_targets = dorothea
)

meta_network_final_rev2 <- meta_network_final_rev2_t2

measurements_rev2 <- filter_inputs(TF_input_COSMOS_inPKN, meta_network_final_rev2)
input_rev2 <- filter_inputs(signaling_input_COSMOS_inPKN, meta_network_final_rev2)
```

## Run CARNIVAL

```{r, eval = FALSE}

# define options for CARNIVAL optimization
timelimit <- 500
mipGAP <- 0.15
#solverPath <- "~/Documents/gsk/COSMOS/CPLEX_Studio1210/cplex/bin/x86-64_osx/cplex"
used_options <- c(as.character(timelimit), as.character(mipGAP), solverPath)

beepr::beep(1)
CARNIVAL_Result_rev2 <- runCARNIVAL(
  solverPath = solverPath,
  netObj = meta_network_final_rev2,
  measObj = measurements_rev2,
  inputObj = input_rev2,
  weightObj = weights_COSMOS,
  timelimit = timelimit,
  mipGAP = mipGAP,
  solver = "cplex",
  dir_name = "/Users/miraburtscher/Documents/gsk/O2_Network_modeling/tmp"
)

COSMOS_object_rev2 <- structure(list(
  meta_network = meta_network_final_rev2,
  tf_regulon = dorothea,
  signaling_data_UP = input_rev2,
  metabolic_data_DOWN = measurements_rev2,
  expression_data = expression_data,
  weightedSIF = as.data.frame(CARNIVAL_Result_rev2$weightedSIF),
  nodeAtt = as.data.frame(CARNIVAL_Result_rev2$nodesAttributes),
  history = c(runID)
),
used_options = used_options,
class = "cosmos_data"
)
beepr::beep(3)
```



```{r, eval = FALSE}
save(COSMOS_object_fw1, COSMOS_object_fw2, COSMOS_object_rev1, COSMOS_object_rev2,
  file = paste0("C:/Users/burtsche/Documents/COSMOS-TPP_paper/results/", runID, ".RData")
)
```

```{r}
```


# Use COSMOS pacakge

DOES NOT WORK

<!-- ## Carnival options -->

<!-- ```{r} -->

<!-- CARNIVAL_options <- cosmosR::default_CARNIVAL_options() -->
<!-- CARNIVAL_options$solverPath <- "/Users/miraburtscher/Documents/gsk/COSMOS/cplex" -->
<!-- # CARNIVAL_options$solver <- "cplex" #or cbc -->
<!-- CARNIVAL_options$solver <- "cplex" #or cbc -->
<!-- CARNIVAL_options$threads <- 1 -->
<!-- CARNIVAL_options$dir_name <- "/Users/miraburtscher/Documents/gsk/O2_Network_modeling/results" -->

<!-- ``` -->

<!-- ## Input -->


<!-- ```{r} -->
<!-- metaPKN  <- read_csv("~/Documents/gsk/COSMOS/COSMOS_2nd_version/support/meta_network_carnival_ready_exch_solved_fullomni_metfiltered.csv") -->


<!-- signaling_input_COSMOS <- t(input_carnival)[,1] -->
<!-- TF_input_COSMOS <- t(measurement_carnival)[,1] -->

<!-- RNA_input_COSMOS <- ttop_RNA$log2FC -->
<!-- names(RNA_input_COSMOS) <- ttop_RNA$ID -->
<!-- RNA_input_COSMOS <- RNA_input_COSMOS[names(RNA_input_COSMOS)!="XNA"] -->
<!-- names(RNA_input_COSMOS) <- as.character(names(RNA_input_COSMOS)) -->
<!-- RNA_input_COSMOS <- RNA_input_COSMOS[!duplicated(names(RNA_input_COSMOS))] -->
<!-- ``` -->

<!-- ## Forward prerun -->

<!-- ```{r} -->
<!-- library(cosmosR) -->
<!-- CARNIVAL_options$timelimit <- 1000 -->
<!-- CARNIVAL_options$mipGAP <- 0.15 -->

<!-- test_for <- preprocess_COSMOS_signaling_to_metabolism(meta_network = metaPKN, -->
<!--                                         signaling_data = signaling_input_COSMOS, -->
<!--                                         metabolic_data = TF_input_COSMOS, -->
<!--                                                       diff_expression_data = RNA_input_COSMOS, -->
<!--                                                       maximum_network_depth = 8, -->
<!--                                                       remove_unexpressed_nodes = TRUE, -->
<!--                                                       CARNIVAL_options = CARNIVAL_options -->
<!--                                                       ) -->

<!-- ``` -->

<!-- ## Forward prerun 2 -->

<!-- ```{r} -->
<!-- # test_for <- preprocess_COSMOS_signaling_to_metabolism(meta_network = metaPKN_exp, -->
<!-- #                                         signaling_data = signaling_input_COSMOS, -->
<!-- #                                         metabolic_data = TF_input_COSMOS, -->
<!-- #                                                       diff_expression_data = RNA_input_COSMOS, -->
<!-- #                                                       maximum_network_depth = 8, -->
<!-- #                                                       remove_unexpressed_nodes = FALSE, -->
<!-- #                                                       CARNIVAL_options = CARNIVAL_options -->
<!-- #                                                       ) -->
<!-- ``` -->

<!-- ## Forward run -->

<!-- ```{r} -->
<!-- CARNIVAL_options$timelimit <- 3600 -->
<!-- CARNIVAL_options$mipGAP <- 0.1 -->

<!-- # die falschen/gefilterten nodes nach dem preprocessing werden zwar von COSMOS ausgegeben und vom PKN entfernt aber nicht vom input -->
<!-- test_for$metabolic_data <- test_for$metabolic_data[!(names(test_for$metabolic_data) %in% c("X7022", "X5452"))] -->
<!-- test_for$signaling_data <- test_for$signaling_data[!(names(test_for$signaling_data) %in% c("X7022", "X5452"))] -->

<!--  test_result_for <- run_COSMOS_signaling_to_metabolism(data = test_for, -->
<!--                                                        CARNIVAL_options = CARNIVAL_options) -->
<!-- ``` -->

<!-- ## Backward prerun -->

<!-- ```{r} -->
<!-- CARNIVAL_options$timelimit <- 500 -->
<!-- CARNIVAL_options$mipGAP <- 0.15 -->
<!-- test_back <- preprocess_COSMOS_metabolism_to_signaling(meta_network = metaPKN, -->
<!--                                         signaling_data = signaling_input_COSMOS, -->
<!--                                         metabolic_data = TF_input_COSMOS, -->
<!--                                         diff_expression_data = RNA_input_COSMOS, -->
<!--                                         maximum_network_depth = 8, -->
<!--                                         remove_unexpressed_nodes = TRUE, -->
<!--                                         filter_tf_gene_interaction_by_optimization = TRUE, #hier hatte ich auch FALSE ausprobiert, hat nichts geandert -->
<!--                                         CARNIVAL_options = CARNIVAL_options -->

<!-- ) -->
<!-- ``` -->

<!-- ## Backward run -->

<!-- ```{r} -->
<!-- CARNIVAL_options$timelimit <- 72000 -->
<!-- CARNIVAL_options$mipGAP <- 0.15 -->

<!-- test_back$metabolic_data <- test_back$metabolic_data[!(names(test_back$metabolic_data) %in% c("X7022"))] -->

<!-- test_back$signaling_data <- test_back$signaling_data[!(names(test_back$signaling_data) %in% c("X7022"))] -->

<!-- test_result_back <- run_COSMOS_metabolism_to_signaling(data = test_back, -->
<!--                                                       CARNIVAL_options = CARNIVAL_options) -->
<!-- ``` -->
<!-- 65% gap -->
