---
title: "O2M1: Check different COSMOS setup strategies"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide
---

This script provides the comparison of networks modeled based on different TPP activity estimation strategies.

# General settings


```{r setup}
knitr::opts_chunk$set(fig.width = 8*(1+sqrt(5))/2, fig.height = 8,#golden ratio) 
                      echo      = TRUE, 
                      warning   = FALSE,
                      message   = FALSE,
                      include   = TRUE,
                      cache     = TRUE,
                      cache.lazy = FALSE,
                      eval      = TRUE) 
knitr::opts_knit$set(root.dir ="/Users/miraburtscher/Documents/gsk/Master_thesis_extended/O2_Network_modeling/")
```

```{r, message=F, warning =F}
library(tidyverse)
library("reshape2")
library(OmnipathR)
library(xlsx)
library(viridis)
library(ggrepel)
options(connectionObserver = NULL)
library(org.Hs.eg.db)
library(visNetwork)
library(knitr)
library(ggpubr)
library(cosmosR)
library(cowplot)
library(scales)

# set some custom functions
mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
```

```{r, message=F, warning =F, cache =F, eval = FALSE}
library(metabaser) 
#install.packages("~/metabase/metabaser_4.3.0.tar.gz", type="source", repos=NULL, INSTALL_opts = "--no-multiarch")
source("C:/Users/mb923573/Documents/metabase/metabase.R")
connect_to_mb()
```

```{r}
load("data/210825_allTPPscores.RData")
load("data/dorothea.RData")
load("data/KSN.RData")

load("data/2021_08_26_08_26_21_COSMOSinput_highconfTPP_viper.RData")

```

```{r}
transcriptomic_regulators <- viper_trans_top$regulator
phosphoproteomic_regulators <- viper_phospho_top$regulator
```


# Functions

Function 1: extract and annotate relevant dfs of COSMOS rsult object for plotting and analysis.

```{r}
network_processing <- function(COSMOS_object_fw2, # result of the first run
                               COSMOS_object_rev2, # result of the second run
                               TPP_input, # TPP proteins used as input
                               dorothea, # TF-target interactions
                               KSN, # Kinase-target interactions
                               all_input_nodes){ # all input nodes
  TFs <- dorothea$source_genesymbol
  Kinases <- KSN$enzyme_genesymbol
  # get all nodes and convert to gene symbols
  fw_nodes <- COSMOS_object_fw2$nodeAtt %>%
  mutate(EntrezID = gsub("X", "", Node))
  rev_nodes <- COSMOS_object_rev2$nodeAtt %>%
  mutate(EntrezID = gsub("X", "", Node))
  all_nodes_entrezid <- unique(c(fw_nodes$EntrezID, rev_nodes$EntrezID))
  convertIDs <-  mapIds(org.Hs.eg.db, all_nodes_entrezid, 'SYMBOL', 'ENTREZID') 
  # get networks, map to gene symbols and combine
  fw_network <- COSMOS_object_fw2$weightedSIF
  fw_network_mapped <- data.frame(from = convertIDs[gsub("X","",fw_network$Node1)],
                                  sign = fw_network$Sign,
                                  to = convertIDs[gsub("X","",fw_network$Node2)]) %>%
                        mutate(smooth = TRUE,
                               dashes = ifelse(sign > 0, FALSE, TRUE))  %>%
                        distinct(from, sign, to, .keep_all = T)
                        # filter out big artificial cluster
                        # filter(from != "TP63")
  rev_network <- COSMOS_object_rev2$weightedSIF
  rev_network_mapped <- data.frame(from = convertIDs[gsub("X","",rev_network$Node1)],
                                  sign = rev_network$Sign,
                                  to = convertIDs[gsub("X","",rev_network$Node2)]) %>%
    mutate(smooth = TRUE,
           dashes = ifelse(sign > 0, FALSE, TRUE)) %>%
    distinct(from, sign, to, .keep_all = T)
  
  combined_network <- rbind(fw_network_mapped, rev_network_mapped)
  # extract activity information
  all_nodes <- rbind(filter(fw_nodes, AvgAct != 0), filter(rev_nodes, AvgAct != 0))%>%
  mutate(activity = ifelse(AvgAct > 0, "active", "inactive"))
  node_activities <- all_nodes$activity
  names(node_activities) <- mapIds(org.Hs.eg.db, all_nodes$EntrezID, 'SYMBOL', 'ENTREZID')
  missing_nodes <- setdiff(unique(c(combined_network$from, combined_network$to)), names(node_activities))
  dummy <- rep("not available", length(missing_nodes))
  names(dummy) <- missing_nodes
  node_activities <- c(node_activities, dummy)
  # map all information to nodes table
  mapped_nodes <- data.frame(id = unique(c(combined_network$from, combined_network$to)),
                             label = unique(c(combined_network$from, combined_network$to))) %>%
                  rowwise() %>%
                  mutate(measured = node_activities[[id]])
  mapped_nodes <- mapped_nodes %>%
    mutate(type = ifelse(id %in% Kinases, "Kinase", "other"),
           type = ifelse(id %in% TFs, "TF", type),
           type = ifelse(id %in% TPP_input, "TPP", type),
           shape = ifelse(id %in% all_input_nodes, "square", "triangle"),
           group = paste(type, measured, sep = "_"),
           title = paste0(label, "_", group))
  # store into visNetwork input object
  meta_data <- c(COSMOS_object_fw2$history, COSMOS_object_rev2$history)
  visNetwork_object <- structure(list(fw_network = fw_network_mapped,
                                      fw_background = COSMOS_object_fw2$meta_network,
                                      rev_network = rev_network_mapped,
                                      rev_background = COSMOS_object_rev2$meta_network,
                                      combined_network = combined_network,
                                      nodes = mapped_nodes,
                                      metadata = meta_data,
                                      class = "visNetworkinput"))
  
}
```

Function 2: extract main graph features of COSMOS networks

```{r}
network_checks <- function(visNetwork_object, # result of function 1
                           TPP_input, 
                           transcriptomic_regulators, 
                           phosphoproteomic_regulators, 
                           dorothea, 
                           KSN){
  # number of nodes and edges
  no_edges <- nrow(as.data.frame(visNetwork_object$combined_network))
  no_nodes <- nrow(as.data.frame(visNetwork_object$nodes))
  # convert to igraph graph object
  graph <-  graph_from_data_frame(as.data.frame(visNetwork_object$combined_network)[,c(1,3)], directed = TRUE, vertices = NULL)
  # edge density
  network_density <- edge_density(graph, loops = TRUE)
  # node degree centrality
  node_degrees <- degree(graph, v = V(graph), mode = "total",loops = TRUE, normalized = FALSE)
  node_degrees <- data.frame(Node = names(node_degrees), degree = node_degrees)
  # adjacency matrix
  adj_matrix <- as_adjacency_matrix(graph, type ="both",attr = NULL, edges = FALSE, names = TRUE)
  adj_matrix_m <- as.matrix(adj_matrix)
  adj_matrix_m[adj_matrix_m == 2] <- 1
  # all PKN nodes mapped to gene name identifiers
  PKN_nodes <- unique(c(visNetwork_object$fw_background[["source"]], visNetwork_object$fw_background[["target"]],
                 visNetwork_object$rev_background[["source"]], visNetwork_object$rev_background[["target"]]))
  PKN_nodes_mapped <- mapIds(org.Hs.eg.db, gsub("X", "", PKN_nodes), 'SYMBOL', 'ENTREZID')
  # get input modeled, background modeled and missed input
  # TPP
  included_TPPhits <- intersect(visNetwork_object$nodes[["id"]], TPP_input)
  missedTPPhits <- setdiff(TPP_input, visNetwork_object$nodes[["id"]])
  filtered_TPP <- setdiff(TPP_input, PKN_nodes_mapped)
  # TF
  included_TFs <- intersect(visNetwork_object$nodes[["id"]], transcriptomic_regulators)
  missedTFs <- setdiff(transcriptomic_regulators, visNetwork_object$nodes[["id"]])
  filtered_TFs <- setdiff(transcriptomic_regulators, PKN_nodes_mapped)
  # Kinases
  included_kinases <- intersect(visNetwork_object$nodes[["id"]], phosphoproteomic_regulators)
  missedkinases <- setdiff(phosphoproteomic_regulators, visNetwork_object$nodes[["id"]])
  filtered_kinases <- setdiff(phosphoproteomic_regulators, PKN_nodes_mapped)
  # summary
  input_nodes <- as.data.frame(visNetwork_object$nodes) %>% filter(shape == "square")
  input_nodes <- input_nodes$id
  additional_nodes <- as.data.frame(visNetwork_object$nodes) %>% filter(shape == "triangle")
  additional_nodes <- additional_nodes$id
  missed_nodes <- c(missedTPPhits, missedkinases, missedTFs)
  summary = data.frame(node= c(input_nodes, additional_nodes, missed_nodes),
                       model = c(rep("input", length(input_nodes)),
                                  rep("additional", length(additional_nodes)),
                                  rep("missed", length(missed_nodes)))) %>%
    mutate( type = ifelse(node %in% KSN$enzyme_genesymbol, "Kinase",  "other"),
            type = ifelse(node %in% dorothea$source_genesymbol, "TF",type),
            type = ifelse(node %in% TPP_input, "TPP", type))
  # build result object
  result <- structure(list(no_edges = no_edges,
                            no_nodes = no_nodes,
                            graph = graph,
                            adj_matrix = adj_matrix,
                           adj_matrix_m = adj_matrix_m,
                            node_degrees =node_degrees,
                            network_density = network_density,
                            overall_node_summary = summary))

}
```

# Networks

## only TFs and kinases

```{r}
load("old/2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK.RData")

all_input_nodes <- c(viper_trans_top$regulator, viper_phospho_top$regulator, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK = visNetwork(res_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK$nodes, distinct(res_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK

netprop_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK <- network_checks(res_2021_08_23_21_52_03_Kinases_vs_TFs_viper_GSK, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## alle TPPs NA

```{r}
load("old/2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK.RData")

all_input_nodes <- c(viper_trans_top$regulator, viper_phospho_top$regulator, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK = visNetwork(res_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK$nodes, distinct(res_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
   visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK

netprop_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK <- network_checks(res_2021_08_25_07_52_25_TPPonlyNAKinvsTF_viper_GSK, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## only TPPs with activity estimate

```{r}
load("old/2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK.RData")

all_input_nodes <- c(viper_trans_top$regulator, viper_phospho_top$regulator, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK = visNetwork(res_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK$nodes, distinct(res_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
   visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK

netprop_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK <- network_checks(res_2021_08_23_14_32_40_KinTPPactest_TF_viper_GSK, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```


## TPP activity estimate + weights

```{r}

load("old/2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK.RData")

all_input_nodes <- c(viper_trans_top$regulator, viper_phospho_top$regulator, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK = visNetwork(res_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK$nodes, distinct(res_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
   visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK

netprop_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK <- network_checks(res_2021_08_13_08_11_57_TPPactestweightedKinasesvsTF_viper_GSK, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## TPP activity estimate and NAs + weights

```{r}

load("old/2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK.RData")

all_input_nodes <- c(viper_trans_top$regulator, viper_phospho_top$regulator, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK = visNetwork(res_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK$nodes, distinct(res_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
   visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK

netprop_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK <- network_checks(res_2021_08_13_11_17_52_TPPactestNAweightedKinasesvsTF_viper_GSK, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```


# Comparison

```{r}
values_types =  c("TPP" = "mediumseagreen",
                  "Kinase" =  "darkcyan", 
                  "TF" = "#8b0a50",
                  "other" = "gray40")

values_TPPset =  c("monotony" = "brown",
                   "Fstatistic" =  "indianred", 
                  "highconfidence" = "lightcoral",
                   "noTPP" = "peachpuff")
levels_TPPset <- c("noTPP", "highconfidence", "Fstatistic","monotony")

values_model =  c("input node modelled" =  "darkred", 
            "input node not modelled" = "indianred",
            "PKN node" = "gray80")


values_PKN =  c("COSMOSnew" =  "maroon", 
            "GSK" = "goldenrod1")

values_networkID =  c("TPP activity estimation plus weights" =  "cyan4",
                      "TPP activity estimation and NaN plus weights" = "deeppink3",
                      "TPP activity estimation" = "goldenrod3",
                      "no TPP" = "darkolivegreen4",
                      "TPP only NaN" = "indianred")

```


### Nodes

```{r}
network_list <- mget(ls(pattern= "netprop"))
names(network_list) <- c("TPP activity estimation plus weights", "TPP activity estimation and NaN plus weights", "TPP activity estimation", "no TPP", "TPP only NaN") 
nodes <- map_dfr(network_list, 'overall_node_summary', .id = "networkID") %>% 
  separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F)

nodes_summary <- nodes %>% 
  filter(model != "missed") %>% 
  group_by(TPP_set, replicate, type) %>% 
  summarize(count = n()) %>% 
  group_by(TPP_set, type) %>% 
  mutate(mean = mean(count), sd =sd(count))
```

#### Barplots

```{r}

p_barplot_nodes <- nodes %>% 
  #filter(model != "missed") %>% 
  ggplot(aes(x = factor(networkID, levels = c("TPP activity estimation plus weights", "TPP activity estimation and NaN plus weights", "TPP activity estimation", "no TPP", "TPP only NaN")), 
             fill = networkID)) +
  geom_bar() +
  scale_fill_manual(values = values_networkID) +
  facet_wrap(~ model) +
  theme_bw(16)+
  #theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
  labs(x = "network", y = "total number of nodes", title = "Number of nodes per network")+
  coord_flip()
p_barplot_nodes

p_TPP_nodes <- nodes %>% 
  filter(type == "TPP" & model == "input") %>% 
  ggplot(aes(x =  factor(networkID, 
                         levels = c("no TPP",  "TPP only NaN", "TPP activity estimation","TPP activity estimation plus weights", "TPP activity estimation and NaN plus weights")))) +
  geom_bar(colour = "darkgray", fill ="darkgray") +
  #scale_fill_manual(values = values_networkID) + 
  scale_y_continuous(breaks= pretty_breaks())+
  theme_cowplot(font_size = 12) +
  #theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  #labs(x = "network", y = "number of TPP proteins", title = "Number of integrated and missed TPP proteins") +
  labs(x = "", y = "number of integrated TPP proteins")+
  coord_flip()
p_TPP_nodes
```

```{r}
p_TPP_nodes <- nodes %>% 
  filter(type == "TPP" & model == "input") %>% 
  mutate(networkID = factor(networkID, levels = c("no TPP",  "TPP only NaN", "TPP activity estimation","TPP activity estimation plus weights", "TPP activity estimation and NaN plus weights")),
         ID = as.numeric(networkID)
  )  %>% 
  ggplot(aes(x =  ID)) +
  geom_bar(colour = "darkgray", fill ="darkgray") +
  #scale_fill_manual(values = values_networkID) + 
  scale_y_continuous(breaks= pretty_breaks())+
  theme_cowplot(font_size = 14) +
  #theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  #labs(x = "network", y = "number of TPP proteins", title = "Number of integrated and missed TPP proteins") +
  labs(x = "modeling approaches", y = "number of \n integrated TPP proteins")
  #coord_flip()
p_TPP_nodes
```


#### Centralities

```{r}
centralities <- map_dfr(network_list, 'node_degrees', .id = "networkID") %>%
  left_join(nodes, by = c("Node"= "node", "networkID"))

p_centralities <- ggplot(centralities, aes(x = networkID, y = degree, colour = type, fill = type)) +
  geom_violin(alpha = 0.4) +
  geom_point(position = position_jitterdodge(0.5)) +
  scale_fill_manual(values = values_types) +
  scale_colour_manual(values = values_types) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1)) +
  facet_wrap(~type, ncol = 4) +
  theme_bw(16)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  labs(x = "TPP set", y = "node degree", title = "Degree centrality per TPP set and protein type")
p_centralities
```

### Heatmap

```{r}
pheatmap_input <- nodes %>%
  filter(model != "missed") %>% 
  mutate(conv = ifelse(model == "additional", -1, 0),
         conv = ifelse(model == "input", 1, conv)) %>% 
  distinct(networkID, node, .keep_all = T)%>%
  acast(networkID ~ node, value.var = "conv", fill = 0)


# check <- pheatmap_input %>% 
#   dplyr::select(networkID, node)
# 
# check[duplicated(check[,1:2]),]

annotation_cols <- data.frame(Node = colnames(pheatmap_input))%>%
  left_join(distinct(nodes, node, type), by = c("Node"= "node")) %>% 
  arrange(desc(type)) %>% 
  distinct(Node, .keep_all = T) %>%
  column_to_rownames("Node")

annotation_rows <- data.frame(networkID = rownames(pheatmap_input))%>%
  left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>%
  column_to_rownames("networkID")


my_palette <- c(colorRampPalette(colors = c("#67001F"))(n = 1),
              "gray91",
              c(colorRampPalette(colors = c("darkgreen"))(n = 1)))

annotation_colors = list(
  type = values_types,
  TPP_set = values_networkID)

pheatmap::pheatmap(pheatmap_input, 
                   show_colnames = F, 
                   annotation_col = annotation_cols,
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   legend_breaks = c(-0.8, 0, 0.8), 
                   legend_labels = c("additional", "missed", "input"), 
                   main = "Comparison of network nodes of different TPP sets")


```

### Edges

```{r}
result_list <- mget(ls(pattern= "^res_.*GSK"))
names(result_list) <- c("TPP activity estimation plus weights", "TPP activity estimation and NaN plus weights", "TPP activity estimation", "no TPP", "TPP only NaN") 

TPP_nodes <- nodes %>% 
  filter(type == "TPP") %>% 
  dplyr::select(networkID, node, type)

edges <- map_dfr(result_list, 'combined_network', .id = "networkID")  %>%
  mutate(edge_combined = paste(from, sign, to, sep = "_")) %>%
  distinct(networkID, edge_combined, .keep_all = T)%>% 
  separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) %>% 
  left_join(TPP_nodes, by=c("networkID", "from" = "node"), suffix = c("", "_source")) %>% 
  left_join(TPP_nodes, by=c("networkID", "to" = "node"), suffix = c("_source", "_target"))


```

```{r}
p_barplot_edges <- edges %>% 
  ggplot(aes(x = networkID,  fill = networkID)) +
  geom_bar(position = position_dodge()) +
  scale_fill_manual(values = values_networkID) +
  theme_bw(16)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  labs(x = "network", y = "total number of edges", title = "Number of edges per network")
p_barplot_edges

```

```{r}
pheatmap_input_edges <- edges %>%
  mutate(sign = as.numeric(sign)) %>%
  acast(networkID ~ edge_combined, value.var = "sign", fill = 0)

# annotation_cols <- data.frame(edge = colnames(pheatmap_input_edges))%>%
#   left_join(distinct(edges, edge_combined, sign), by = c("edge"= "edge_combined")) %>%
#   column_to_rownames("edge")

annotation_rows <- data.frame(networkID = rownames(pheatmap_input))%>%
  left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>%
  column_to_rownames("networkID")


my_palette <- c(colorRampPalette(colors = c("#67001F"))(n = 1),
              "gray91",
              c(colorRampPalette(colors = c("darkgreen"))(n = 1)))

annotation_colors = list(
  #type = c(Kinase="red", TF="yellow", other = "blue", TPP = "green"),
  TPP_set = values_networkID)

pheatmap::pheatmap(pheatmap_input_edges, 
                   show_colnames = F, 
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   legend_breaks = c(-0.8, 0, 0.8), 
                   legend_labels = c("inhibition", "missed", "activation"),
                   main = "Comparison of network edges of different TPP sets")

```
### Networks

```{r}
densities <- map(network_list, 'network_density', .id = "networkID") %>% 
  unlist()

densities <- data.frame(networkID = names(densities), density = unname(densities))%>% 
  separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F)

p_densities <- ggplot(densities, aes(x=networkID, y = density, fill = networkID)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = values_networkID) +
  theme_bw(16)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+
  labs(x = "network", y = "density", title = "Comparison of network densities for different TPP sets")
p_densities
```

```{r}

```

