---
title: "O2M2: Analysis and exploration of result networks"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide
---

# General settings


```{r setup}
knitr::opts_chunk$set(echo      = TRUE, 
                      warning   = FALSE,
                      message   = FALSE,
                      include   = TRUE,
                      cache     = TRUE,
                      cache.lazy = FALSE,
                      eval      = TRUE,
                      fig.width = 7*(1+sqrt(5))/2, fig.height = 7) 
knitr::opts_knit$set(root.dir ="C:/Users/burtsche/Documents/COSMOS-TPP_paper")
```

## Packages

```{r, message=F, warning =F}
library(tidyverse)
library(igraph)
library("reshape2")
library(OmnipathR)
library(viridis)
library(ggrepel)
options(connectionObserver = NULL)
library(org.Hs.eg.db)
library(visNetwork)
library(knitr)
library(ggpubr)
#library(NetworkDistance)
library(Rcpp)
library(RColorBrewer)
library(networkD3)
library(kableExtra)
library(ReactomePA)


mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
```

Metabase connection

```{r, message=F, warning =F, cache =F}
# library(metabaser) 
# # install.packages("~/metabase/metabaser_4.3.0.tar.gz", type="source", repos=NULL, INSTALL_opts = "--no-multiarch")
# source("C:/Users/mb923573/Documents/metabase/metabase.R")
# connect_to_mb()
```

## Input

Load data.

```{r}
# TF-target and Kinase-psite interactions
load("data/dorothea.RData")
load("data/KSN.RData")

# COSMOS input objects
load("results/220512_cosmosinput_merged.RData")

# TPP scores
load("results/220530_TPPallscores.RData")
```

Set transcriptomic and phosphoproteomic regulators (from input objects).

```{r}
transcriptomic_regulators <- viper_trans_top$regulator
phosphoproteomic_regulators <- viper_phospho_top$regulator
```

## Functions

Function 1: extract and annotate relevant dfs of COSMOS rsult object for plotting and analysis.

```{r}
network_processing <- function(COSMOS_object_fw2, # result of the first run
                               COSMOS_object_rev2, # result of the second run
                               TPP_input, # TPP proteins used as input
                               dorothea, # TF-target interactions
                               KSN, # Kinase-target interactions
                               all_input_nodes){ # all input nodes
  TFs <- dorothea$source_genesymbol
  Kinases <- KSN$enzyme_genesymbol
  # get all nodes and convert to gene symbols
  fw_nodes <- COSMOS_object_fw2$nodeAtt %>%
  mutate(EntrezID = gsub("X", "", Node))
  rev_nodes <- COSMOS_object_rev2$nodeAtt %>%
  mutate(EntrezID = gsub("X", "", Node))
  all_nodes_entrezid <- unique(c(fw_nodes$EntrezID, rev_nodes$EntrezID))
  convertIDs <-  mapIds(org.Hs.eg.db, all_nodes_entrezid, 'SYMBOL', 'ENTREZID') 
  # get networks, map to gene symbols and combine
  fw_network <- COSMOS_object_fw2$weightedSIF
  fw_network_mapped <- data.frame(from = convertIDs[gsub("X","",fw_network$Node1)],
                                  sign = fw_network$Sign,
                                  to = convertIDs[gsub("X","",fw_network$Node2)]) %>%
                        mutate(smooth = TRUE,
                               dashes = ifelse(sign > 0, FALSE, TRUE))  %>%
                        distinct(from, sign, to, .keep_all = T)
                        # filter out big artificial cluster
                        # filter(from != "TP63")
  rev_network <- COSMOS_object_rev2$weightedSIF
  rev_network_mapped <- data.frame(from = convertIDs[gsub("X","",rev_network$Node1)],
                                  sign = rev_network$Sign,
                                  to = convertIDs[gsub("X","",rev_network$Node2)]) %>%
    mutate(smooth = TRUE,
           dashes = ifelse(sign > 0, FALSE, TRUE)) %>%
    distinct(from, sign, to, .keep_all = T)
  
  combined_network <- rbind(fw_network_mapped, rev_network_mapped)
  # extract activity information
  all_nodes <- rbind(filter(fw_nodes, AvgAct != 0), filter(rev_nodes, AvgAct != 0))%>%
  mutate(activity = ifelse(AvgAct > 0, "active", "inactive"))
  node_activities <- all_nodes$activity
  names(node_activities) <- mapIds(org.Hs.eg.db, all_nodes$EntrezID, 'SYMBOL', 'ENTREZID')
  missing_nodes <- setdiff(unique(c(combined_network$from, combined_network$to)), names(node_activities))
  dummy <- rep("not available", length(missing_nodes))
  names(dummy) <- missing_nodes
  node_activities <- c(node_activities, dummy)
  # map all information to nodes table
  mapped_nodes <- data.frame(id = unique(c(combined_network$from, combined_network$to)),
                             label = unique(c(combined_network$from, combined_network$to))) %>%
                  rowwise() %>%
                  mutate(measured = node_activities[[id]])
  mapped_nodes <- mapped_nodes %>%
    mutate(type = ifelse(id %in% Kinases, "Kinase", "other"),
           type = ifelse(id %in% TFs, "TF", type),
           type = ifelse(id %in% TPP_input, "TPP", type),
           shape = ifelse(id %in% all_input_nodes, "square", "triangle"),
           group = paste(type, measured, sep = "_"),
           title = paste0(label, "_", group))
  # store into visNetwork input object
  meta_data <- c(COSMOS_object_fw2$history, COSMOS_object_rev2$history)
  visNetwork_object <- structure(list(fw_network = fw_network_mapped,
                                      fw_background = COSMOS_object_fw2$meta_network,
                                      rev_network = rev_network_mapped,
                                      rev_background = COSMOS_object_rev2$meta_network,
                                      combined_network = combined_network,
                                      nodes = mapped_nodes,
                                      metadata = meta_data,
                                      class = "visNetworkinput"))
  
}
```

Function 2: extract main graph features of COSMOS networks

```{r}
network_checks <- function(visNetwork_object, # result of function 1
                           TPP_input, 
                           transcriptomic_regulators, 
                           phosphoproteomic_regulators, 
                           dorothea, 
                           KSN){
  # number of nodes and edges
  no_edges <- nrow(as.data.frame(visNetwork_object$combined_network))
  no_nodes <- nrow(as.data.frame(visNetwork_object$nodes))
  # convert to igraph graph object
  graph <-  graph_from_data_frame(as.data.frame(visNetwork_object$combined_network)[,c(1,3)], directed = TRUE, vertices = NULL)
  # edge density
  network_density <- edge_density(graph, loops = TRUE)
  # node degree centrality
  node_degrees <- degree(graph, v = V(graph), mode = "total",loops = TRUE, normalized = FALSE)
  node_degrees <- data.frame(Node = names(node_degrees), degree = node_degrees)
  # adjacency matrix
  adj_matrix <- as_adjacency_matrix(graph, type ="both",attr = NULL, edges = FALSE, names = TRUE)
  adj_matrix_m <- as.matrix(adj_matrix)
  adj_matrix_m[adj_matrix_m == 2] <- 1
  # all PKN nodes mapped to gene name identifiers
  PKN_nodes <- unique(c(visNetwork_object$fw_background[["source"]], visNetwork_object$fw_background[["target"]],
                 visNetwork_object$rev_background[["source"]], visNetwork_object$rev_background[["target"]]))
  PKN_nodes_mapped <- mapIds(org.Hs.eg.db, gsub("X", "", PKN_nodes), 'SYMBOL', 'ENTREZID')
  # get input modeled, background modeled and missed input
  # TPP
  included_TPPhits <- intersect(visNetwork_object$nodes[["id"]], TPP_input)
  missedTPPhits <- setdiff(TPP_input, visNetwork_object$nodes[["id"]])
  filtered_TPP <- setdiff(TPP_input, PKN_nodes_mapped)
  # TF
  included_TFs <- intersect(visNetwork_object$nodes[["id"]], transcriptomic_regulators)
  missedTFs <- setdiff(transcriptomic_regulators, visNetwork_object$nodes[["id"]])
  filtered_TFs <- setdiff(transcriptomic_regulators, PKN_nodes_mapped)
  # Kinases
  included_kinases <- intersect(visNetwork_object$nodes[["id"]], phosphoproteomic_regulators)
  missedkinases <- setdiff(phosphoproteomic_regulators, visNetwork_object$nodes[["id"]])
  filtered_kinases <- setdiff(phosphoproteomic_regulators, PKN_nodes_mapped)
  # summary
  input_nodes <- as.data.frame(visNetwork_object$nodes) %>% filter(shape == "square")
  input_nodes <- input_nodes$id
  additional_nodes <- as.data.frame(visNetwork_object$nodes) %>% filter(shape == "triangle")
  additional_nodes <- additional_nodes$id
  missed_nodes <- c(missedTPPhits, missedkinases, missedTFs)
  summary = data.frame(node= c(input_nodes, additional_nodes, missed_nodes),
                       model = c(rep("input", length(input_nodes)),
                                  rep("additional", length(additional_nodes)),
                                  rep("missed", length(missed_nodes)))) %>%
    mutate( type = ifelse(node %in% KSN$enzyme_genesymbol, "Kinase",  "other"),
            type = ifelse(node %in% dorothea$source_genesymbol, "TF",type),
            type = ifelse(node %in% TPP_input, "TPP", type))
  # build result object
  result <- structure(list(no_edges = no_edges,
                            no_nodes = no_nodes,
                            graph = graph,
                            adj_matrix = adj_matrix,
                           adj_matrix_m = adj_matrix_m,
                            node_degrees =node_degrees,
                            network_density = network_density,
                            overall_node_summary = summary))

}
```

Function 3: Pathway enrichment with metabase

```{r}
GSE_RPA <- function(geneList, universe) {
  geneList <- mapIds(org.Hs.eg.db, geneList, "ENTREZID", "SYMBOL")
  universe <- mapIds(org.Hs.eg.db, universe,'ENTREZID','SYMBOL')
  pathway_enrichment <- enrichPathway(
    gene = geneList,
    organism = "human",
    pvalueCutoff = 1,
    pAdjustMethod = "BH",
    qvalueCutoff = 1,
    universe = universe,
    minGSSize = 5,
    maxGSSize = 500,
    readable = TRUE
  )
}
```


# Result networks with COSMOSnew PKN (latest COSMOS setup)

## no TPP

Networks modeled from kinases to TFs and the other way round without TPP proteins. Top 30 kinases and top 30 TFs of footprinting analysis (NES).

### 2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1

```{r}
load("results/2022_05_30_07_42_19_testforpaper_onThinkPAd_V1.RData")

all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators)
noTPP <- c()


res_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1 = visNetwork(res_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1$nodes, distinct(res_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1

netprop_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1 <- network_checks(res_2021_08_22_10_04_11_noTPP_viper_COSMOSnew_rep1, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2

```{r}
load("results/2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2.RData")

res_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2 = visNetwork(res_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2$nodes, distinct(res_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2

netprop_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2 <- network_checks(res_2021_08_22_21_30_33_noTPP_viper_COSMOSnew_rep2, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3

```{r}
load("results/2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3.RData")


res_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3 = visNetwork(res_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3$nodes, distinct(res_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3

netprop_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3 <- network_checks(res_2021_08_23_07_36_48_noTPP_viper_COSMOSnew_rep3, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## F-statistic based input

TPP proteins identified with a F-statistic based approach. Kinases and TFs same as before.

### 2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1

```{r}
load("results/2022_05_30_07_42_19_testforpaper_onThinkPAd_V1.RData")

all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final$TPP_protein)

res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1= network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1 = visNetwork(res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$nodes, distinct(res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1

netprop_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1 <- network_checks(res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1, TPP_cosmos_input_final$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

#### DNA damage checkpoint example

```{r}
interesting_pathwaygenes <- c("ATR", "CHEK2, CNNA2", "FOXM1", "ATM", "CDK1")
 
filtered_network <- res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$combined_network %>%
  as.data.frame() %>%
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  dplyr::mutate(color = "black")

filtered_nodes <- res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$nodes %>%
  as.data.frame() %>%
  filter(id %in% filtered_network$from | id %in% filtered_network$to) %>% 
  mutate(group = ifelse(id == "CDK2", "TPP_active", group))
  
test = visNetwork(filtered_nodes , distinct(filtered_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '35px arial black bold') %>%
    #visLegend() %>%
    #visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)


test
```

### 2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2

```{r}
load("results/2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2.RData")

res_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2= network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_Fstat$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2 = visNetwork(res_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2$nodes, distinct(res_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2

netprop_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2 <- network_checks(res_2021_08_26_13_46_35_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep2, TPP_cosmos_input_final_Fstat$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3

```{r}
load("results/2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3.RData")

res_2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3= network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_Fstat$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3 = visNetwork(res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$nodes, distinct(res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3

netprop_2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3 <- network_checks(res_2021_08_26_15_29_33_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep3, TPP_cosmos_input_final_Fstat$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

# Result networks with GSK PKN (latest COSMOS setup)

## no TPP

Networks modeled from kinases to TFs and the other way round without TPP proteins. Top 30 kinases and top 30 TFs of footprinting analysis (NES).

### 2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1

```{r}
load("results/2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1.RData")

all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators)
noTPP <- c()

res_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1 = visNetwork(res_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1$nodes, distinct(res_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1

netprop_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1 <- network_checks(res_2021_09_09_12_03_31_noTPP_KinvsTF_GSK_rep1, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2

```{r}
load("results/2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2.RData")


res_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2 = visNetwork(res_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2$nodes, distinct(res_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2

netprop_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2 <- network_checks(res_2021_09_09_15_08_40_noTPP_KinvsTF_GSK_rep2, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3

```{r}
load("results/2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3.RData")

res_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, noTPP, dorothea, KSN, all_input_nodes)

p_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3 = visNetwork(res_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3$nodes, distinct(res_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visLayout(randomSeed = 123)
p_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3

netprop_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3 <- network_checks(res_2021_09_09_17_51_40_noTPP_KinvsTF_GSK_rep3, noTPP, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## High confidence TPP input

TPP proteins which were found in 3 out of 4 hit calling strategies. Kinases and TFs same as before.

### 2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1

```{r}
# load("results/2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK_sparseinput.RData")
# 
# 
# all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_highconf$TPP_protein)
# 
# res_2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)
# 
# p_2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK = visNetwork(res_2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK$nodes, distinct(res_2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK$combined_network)) %>%
#     visGroups(groupname = "TPP_active", color = "seagreen") %>%
#     visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
#     visGroups(groupname = "TF_active", color = "darkred") %>%
#     visGroups(groupname = "TF_inactive", color = "orangered") %>%
#     visGroups(groupname = "Kinase_active", color = "navy") %>%
#     visGroups(groupname = "Kinase_inactive", color = "lightskyblue") %>%
#     visGroups(groupname = "other_active", color = "gray") %>%
#     visGroups(groupname = "other_inactive", color = "gainsboro") %>%
#     # standard options
#     visEdges(arrows = "to") %>%
#     visLegend() %>%
#     visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
#     visLayout(randomSeed = 123)
# p_2021_08_16_09_42_26_TPPcorrNAweight_adjREV_viper_GSK

```


```{r}
load("results/2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1.RData")


all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_highconf$TPP_protein)

res_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1 = visNetwork(res_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1$nodes, distinct(res_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1

netprop_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1 <- network_checks(res_2021_09_03_14_40_10_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep1, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2


```{r}
load("results/2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2.RData")

res_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2 = visNetwork(res_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2$nodes, distinct(res_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2

netprop_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2 <- network_checks(res_2021_09_04_01_12_29_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep2, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3

```{r}
load("results/2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3.RData")

res_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_highconf$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3 = visNetwork(res_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3$nodes, distinct(res_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3

netprop_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3 <- network_checks(res_2021_09_06_08_31_37_TPPactestNA_NAweight_highconf_adjrev_viper_GSK_rep3, TPP_cosmos_input_final_highconf$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

## Monotony score based input

TPP proteins detected as hit via the monotony score. Kinases and TFs same as before.

### 2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2

```{r}
load("results/2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2.RData")
COSMOS_object_rev2 <-COSMOS_object_rev1

all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_monotony$TPP_protein)

res_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_monotony$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2 = visNetwork(res_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2$nodes, (res_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2

netprop_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2 <- network_checks(res_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2, TPP_cosmos_input_final_monotony$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

FZR1

Substrate-specific adapter for the anaphase promoting complex/cyclosome (APC/C) E3 ubiquitin-protein ligase complex. Associates with the APC/C in late mitosis, in replacement of CDC20, and activates the APC/C during anaphase and telophase. The APC/C remains active in degrading substrates to ensure that positive regulators of the cell cycle do not accumulate prematurely. At the G1/S transition FZR1 is phosphorylated, leading to its dissociation from the APC/C. Following DNA damage, it is required for the G2 DNA damage checkpoint: its dephosphorylation and reassociation with the APC/C leads to the ubiquitination of PLK1, preventing entry into mitosis. Acts as an adapter for APC/C to target the DNA-end resection factor RBBP8/CtIP for ubiquitination and subsequent proteasomal degradation. Through the regulation of RBBP8/CtIP protein turnover, may play a role in DNA damage response, favoring DNA double-strand repair through error-prone non-homologous end joining (NHEJ) over error-free, RBBP8-mediated homologous recombination (HR) (PubMed:25349192).

YY1

Involved in DNA repair. In vitro, binds to DNA recombination intermediate structures (Holliday junctions). Proposed core component of the chromatin remodeling INO80 complex which is involved in transcriptional regulation, DNA replication and probably DNA repair; proposed to target the INO80 complex to YY1-responsive elements.
"A YY1-INO80 complex regulates genomic stability through homologous recombination-based repair."

### 2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2

```{r}
load("results/2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2.RData")

res_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2= network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_monotony$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2 = visNetwork(res_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2$nodes, distinct(res_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2

netprop_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2 <- network_checks(res_2021_09_08_15_37_46_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep2_norev2, TPP_cosmos_input_final_monotony$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2

```{r}
load("results/2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2.RData")

all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_monotony$TPP_protein)

res_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_monotony$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2 = visNetwork(res_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2$nodes, distinct(res_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr") %>% 
  visOptions(manipulation = TRUE)
p_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2

netprop_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2 <- network_checks(res_2021_09_09_06_22_03_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep3_norev2, TPP_cosmos_input_final_monotony$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

<!-- ### 2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper_GSK_UWB -->

<!-- ```{r} -->
<!-- load("results/2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper_GSK_UWB.RData") -->


<!-- all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_monotony$TPP_protein) -->

<!-- res_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_monotony$TPP_protein, dorothea, KSN, all_input_nodes) -->

<!-- p_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper = visNetwork(res_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper$nodes, distinct(res_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper$combined_network)) %>% -->
<!--     visGroups(groupname = "TPP_active", color = "mediumseagreen") %>% -->
<!--     visGroups(groupname = "TPP_inactive", color = "palegreen") %>% -->
<!--     visGroups(groupname = "TF_active", color = "#8b0a50") %>% -->
<!--     visGroups(groupname = "TF_inactive", color = "palevioletred") %>% -->
<!--     visGroups(groupname = "Kinase_active", color = "darkcyan") %>% -->
<!--     visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>% -->
<!--     visGroups(groupname = "other_active", color = "gray") %>% -->
<!--     visGroups(groupname = "other_inactive", color = "gainsboro") %>% -->
<!--     # standard options -->
<!--     visEdges(arrows = "to") %>% -->
<!--     visLegend() %>% -->
<!--     visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>% -->
<!--     visNetwork::visIgraphLayout(layout =  "layout_with_fr") -->
<!-- p_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper -->

<!-- ``` -->

<!-- # ```{r} -->
<!-- # netprop_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper <- network_checks(res_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper, TPP_cosmos_input_final_monotony$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN) -->
<!-- # ``` -->

## F-statistic based input

TPP proteins identified with a F-statistic based approach. Kinases and TFs same as before.

### 2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1

```{r}
load("results/2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1.RData")


all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_Fstat$TPP_protein)

res_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_Fstat$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1 = visNetwork(res_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1$nodes, distinct(res_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)  %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1

netprop_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1 <- network_checks(res_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1, TPP_cosmos_input_final_Fstat$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

UBB

Ubiquitin:
Exists either covalently attached to another protein, or free (unanchored). When covalently bound, it is conjugated to target proteins via an isopeptide bond either as a monomer (monoubiquitin), a polymer linked via different Lys residues of the ubiquitin (polyubiquitin chains) or a linear polymer linked via the initiator Met of the ubiquitin (linear polyubiquitin chains). Polyubiquitin chains, when attached to a target protein, have different functions depending on the Lys residue of the ubiquitin that is linked: Lys-6-linked may be involved in DNA repair; Lys-11-linked is involved in ERAD (endoplasmic reticulum-associated degradation) and in cell-cycle regulation; Lys-29-linked is involved in lysosomal degradation; Lys-33-linked is involved in kinase modification; Lys-48-linked is involved in protein degradation via the proteasome; Lys-63-linked is involved in endocytosis, DNA-damage responses as well as in signaling processes leading to activation of the transcription factor NF-kappa-B. Linear polymer chains formed via attachment by the initiator Met lead to cell signaling. Ubiquitin is usually conjugated to Lys residues of target proteins, however, in rare cases, conjugation to Cys or Ser residues has been observed. When polyubiquitin is free (unanchored-polyubiquitin), it also has distinct roles, such as in activation of protein kinases, and in signaling.

LON9

Acts as a tumor suppressor. Inhibits DNA synthesis. Its ability to inhibit oncogenic transformation is mediated through its association with RB1. Plays a role in the expression of genes required for the G1/S transition.


### 2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2

```{r}
load("results/2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2.RData")


all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_Fstat$TPP_protein)

res_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_Fstat$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2 = visNetwork(res_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2$nodes, distinct(res_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2

netprop_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2 <- network_checks(res_2021_09_05_18_59_21_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep2, TPP_cosmos_input_final_Fstat$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```

### 2021_08_31_17_42_34_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3

```{r}
load("results/2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3.RData")


all_input_nodes <- c(transcriptomic_regulators, phosphoproteomic_regulators, TPP_cosmos_input_final_Fstat$TPP_protein)

res_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3 = network_processing(COSMOS_object_fw2, COSMOS_object_rev2, TPP_cosmos_input_final_Fstat$TPP_protein, dorothea, KSN, all_input_nodes)

p_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3 = visNetwork(res_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3$nodes, distinct(res_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3$combined_network)) %>%
    visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    visGroups(groupname = "other_active", color = "gray") %>%
    visGroups(groupname = "other_inactive", color = "gainsboro") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)%>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr")
p_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3

netprop_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3 <- network_checks(res_2021_09_06_01_41_33_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep3, TPP_cosmos_input_final_Fstat$TPP_protein, transcriptomic_regulators, phosphoproteomic_regulators, dorothea, KSN)
```


```{r}
# save(res_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2,
#      netprop_2021_09_06_15_16_49_TPPactestNA_NAweight_mono_adjrev_viper_GSK_rep1_norev2,
#      res_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1,
#      netprop_2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1,
#      res_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper,
#      netprop_2021_09_12_17_30_06_TPPactestNA_newNAweight_mono_adjrev_viper,
#      file = "results/2021-09-13_UWB_networks.RData")
```


# Comparison of COSMOSnew PKN and GSK PKN results combined

## Set colourscales

```{r}
# Networks
    # visGroups(groupname = "TPP_active", color = "mediumseagreen") %>%
    # visGroups(groupname = "TPP_inactive", color = "palegreen") %>%
    # visGroups(groupname = "TF_active", color = "#8b0a50") %>%
    # visGroups(groupname = "TF_inactive", color = "palevioletred") %>%
    # visGroups(groupname = "Kinase_active", color = "darkcyan") %>%
    # visGroups(groupname = "Kinase_inactive", color = "mediumturquoise") %>%
    # visGroups(groupname = "other_active", color = "gray") %>%
    # visGroups(groupname = "other_inactive", color = "gainsboro") %>%

values_types =  c("TPP" = "mediumseagreen",
                  "Kinase" =  "darkcyan", 
                  "TF" = "#8b0a50",
                  "other" = "gray40")

values_TPPset =  c("monotony" = "brown",
                   "Fstatistic" =  "indianred", 
                  "highconfidence" = "lightcoral",
                   "noTPP" = "peachpuff")
levels_TPPset <- c("noTPP", "highconfidence", "Fstatistic","monotony")

values_TPPset_paper =  c("large" = "brown",
                   "medium" =  "indianred", 
                  "small" = "lightcoral",
                   "noTPP" = "peachpuff")

values_model =  c("input node modelled" =  "darkred", 
            "input node not modelled" = "indianred",
            "PKN node" = "gray80")


values_PKN =  c("COSMOSnew" =  "maroon", 
            "GSK" = "goldenrod1")

values_PKN_paper =  c("PKN2" =  "maroon", 
            "PKN1" = "goldenrod1")

values_networkID =  c("Fstat_rep1" =  "cyan4",
                      "Fstat_rep2" =  "cyan3",
                      "Fstat_rep3" =  "cyan2", 
                      "highconf_rep1" = "deeppink3",
                      "highconf_rep2" = "deeppink2",
                      "highconf_rep3" = "deeppink1",
                      "mono_rep1" = "goldenrod3",
                      "mono_rep2" = "goldenrod2",
                      "mono_rep3" = "goldenrod1",
                      "noTPP_rep1" = "darkolivegreen4",
                      "noTPP_rep2" = "darkolivegreen3",
                      "noTPP_rep3" = "darkolivegreen2")

```

## Extract relevant data frames from result objects

Nodes

```{r}
result_list <- mget(ls(pattern= "^res_"))
names(result_list) <- c("noTPP_rep1_COSMOSnew", "noTPP_rep2_COSMOSnew", "noTPP_rep3_COSMOSnew", 
                        "Fstatistic_rep1_COSMOSnew", "Fstatistic_rep2_COSMOSnew", "Fstatistic_rep3_COSMOSnew", 
                        "highconfidence_rep1_COSMOSnew", "monotony_rep1_COSMOSnew", "monotony_rep2_COSMOSnew", 
                        "monotony_rep3_COSMOSnew", "highconfidence_rep1_GSK","highconfidence_rep2_GSK",
                        "Fstatistic_rep1_GSK", "Fstatistic_rep2_GSK", "Fstatistic_rep3_GSK",
                        "highconfidence_rep3_GSK","monotony_rep1_GSK", "monotony_rep2_GSK",
                        "monotony_rep3_GSK","noTPP_rep1_GSK", "noTPP_rep2_GSK", "noTPP_rep3_GSK") 

nodes <- map_dfr(result_list, 'nodes', .id = "networkID")%>% 
  dplyr::select(networkID, label, measured)
```

Node attributes

```{r}
network_list <- mget(ls(pattern= "^netprop_"))
names(network_list) <- c("noTPP_rep1_COSMOSnew", "noTPP_rep2_COSMOSnew", "noTPP_rep3_COSMOSnew", 
                        "Fstatistic_rep1_COSMOSnew", "Fstatistic_rep2_COSMOSnew", "Fstatistic_rep3_COSMOSnew", 
                        "highconfidence_rep1_COSMOSnew", "monotony_rep1_COSMOSnew", "monotony_rep2_COSMOSnew", 
                        "monotony_rep3_COSMOSnew", "highconfidence_rep1_GSK","highconfidence_rep2_GSK",
                        "Fstatistic_rep1_GSK", "Fstatistic_rep2_GSK", "Fstatistic_rep3_GSK",
                        "highconfidence_rep3_GSK","monotony_rep1_GSK", "monotony_rep2_GSK",
                        "monotony_rep3_GSK","noTPP_rep1_GSK", "noTPP_rep2_GSK", "noTPP_rep3_GSK") 

nodes <- map_dfr(network_list, 'overall_node_summary', .id = "networkID") %>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F) %>% 
  left_join(nodes, by = c("networkID", "node" = "label")) %>% 
  mutate(model = gsub("input", "input node modelled", model),
         model = gsub("missed", "input node not modelled", model),
          model = gsub("additional", "PKN node", model))
```

```{r, eval=FALSE}
save(result_list, network_list, file = "results/2022_02_17_MasterThesis_networks.RData")
```


## Analysis of Nodes

```{r, fig.width=7, fig.height=6}
nodes_summary <- nodes %>% 
  filter(model != "input node not modelled") %>% 
  group_by(PKN, TPP_set, replicate, type) %>% 
  summarize(count = n()) %>% 
  group_by(PKN, TPP_set, type) %>% 
  mutate(mean = mean(count), sd =sd(count)) %>% 
  bind_rows(data.frame( type = c("TPP", "TPP"),
                        PKN = c("GSK", "COSMOSnew"),
                       TPP_set = c("noTPP", "noTPP"),
                       mean = c(0,0), sd = c(NA, NA)))

p_barplot_nodes_sum <- nodes_summary %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = type,
             colour = type)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = type), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line(size =1 ) +
  scale_colour_manual(values = values_types) +
  facet_wrap(~PKN) +
  theme_bw(14) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank(),
        legend.position = "top") +
  labs(x = "", y = "number of nodes", title = "Mean number of nodes per node type and TPP set")
p_barplot_nodes_sum
```

Thesis plot

```{r, eval = F}
p_barplot_nodes_sum <- nodes_summary %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = type,
             colour = type)) +
  #geom_bar(stat = "identity",position = position_dodge(), fill = "black") +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = type), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line(size =1 ) +
  scale_colour_manual(values = values_types) +
  facet_wrap(~PKN) +
  theme_bw(16) +
   theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank()) +
  #labs(x = "TPP set", y = "total number of nodes", title = "Mean number of nodes per node type and TPP set") 
  labs(x = "", y = "number of nodes")
p_barplot_nodes_sum
```

```{r, fig.width=7, fig.height=6}
nodes_summary <- nodes %>% 
  group_by(PKN, TPP_set, replicate, model) %>% 
  summarize(count = n()) %>% 
  group_by(PKN, TPP_set, model) %>% 
  mutate(mean = mean(count), sd =sd(count))

p_barplot_nodes <- nodes_summary %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = model,
             colour = model)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = model), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  scale_colour_manual(values = values_model) +
  facet_wrap(~PKN) +
  theme_bw(14)+
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank(),
        legend.position = "top") +
  labs(x = "", y = "number of nodes", title = "Mean number of nodes per modeling type and TPP set")
p_barplot_nodes

```

Thesis plot

```{r, eval = F}
p_barplot_nodes <- nodes_summary %>% 
  #filter(model != "missed") %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = model,
             colour = model)) +
  #geom_bar(stat = "identity",position = position_dodge(), fill = "black") +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = model), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line(size = 1) +
  #scale_fill_manual(values = values_TPPset) +
  scale_colour_manual(values = values_model) +
  #facet_grid(PKN ~ model) +
  facet_wrap(~PKN) +
  theme_bw(16)+
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank()) +
  #labs(x = "network", y = "total number of nodes", title = "Number of nodes per network") 
  labs(x = "", y = "number of nodes")
p_barplot_nodes
```

### Centralities

```{r, fig.width=7, fig.height=6}
centralities <- map_dfr(network_list, 'node_degrees', .id = "networkID") %>%
  left_join(nodes, by = c("Node"= "node", "networkID")) %>% 
  group_by(PKN, TPP_set, type, Node) %>% 
  summarise(mean = mean(degree), sd= sd(degree)) %>% 
  mutate(mean_adj = ifelse(mean < 5, NA, mean))

pos <- position_jitter(seed = 2, width = 0.2)
p_centralities <- ggplot(centralities, aes(x = factor(TPP_set, levels = levels_TPPset), 
                                           y = mean, colour = type, fill = type))  +
  geom_violin(position = position_dodge(width = 0.8), alpha = 0.2) +
  geom_jitter(aes(y = mean_adj), position = pos) +
  geom_text_repel(aes(label = ifelse(mean > 5,Node, "")), position = pos, size = 3, 
                  colour = "black", max.overlaps = 8, min.segment.length = 0.1) +
  scale_fill_manual(values = values_types) +
  scale_colour_manual(values = values_types) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1)) +
  facet_grid(PKN~type, scales = "free_x") +
  theme_bw(14)+
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(),
        legend.position = "none") +
  labs(x = "", y = "node degree", title = "Degree centrality per TPP set and protein type")
p_centralities
```

Thesis plot

```{r, eval = F}
pos <- position_jitter(seed = 2, width = 0.2)
p_centralities <- ggplot(centralities, aes(x = factor(TPP_set, levels = levels_TPPset), 
                                           y = mean, colour = type, fill = type))  +
  geom_violin(position = position_dodge(width = 0.8), alpha = 0.2) +
  geom_jitter(aes(y = mean_adj), position = pos) +
  geom_text_repel(aes(label = ifelse(mean > 5,Node, "")), position = pos, size = 3, 
                  colour = "black", max.overlaps = 8, min.segment.length = 0.1) +
  scale_fill_manual(values = values_types) +
  scale_colour_manual(values = values_types) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1)) +
  facet_grid(PKN~type, scales = "free_x") +
  theme_bw(16)+
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank(), panel.grid.major.x = element_blank()) +
  #labs(x = "TPP set", y = "node degree", title = "Degree centrality per TPP set and protein type")
  labs(x = "", y = "node degree")
p_centralities
```

### Heatmap

Clustering network replicates based on node activities 

```{r}
pheatmap_input1_COSMOS <- nodes %>%
  filter(model != "missed") %>% 
  mutate(conv = ifelse(model == "additional", -1, 0),
         conv = ifelse(model == "input", 1, conv)) %>% 
  distinct(networkID, node, .keep_all = T)%>%
  acast(networkID ~ node, value.var = "conv", fill = 0)

pheatmap_input2_COSMOS <- nodes %>%
  filter(model != "missed") %>% 
  mutate(conv = ifelse(measured == "inactive", -1, 0),
         conv = ifelse(measured == "active", 1, conv)) %>% 
  distinct(networkID, node, .keep_all = T)%>%
  acast(networkID ~ node, value.var = "conv", fill = 0)

annotation_cols <- data.frame(Node = colnames(pheatmap_input1_COSMOS))%>%
  left_join(distinct(nodes, node, type), by = c("Node"= "node")) %>% 
  arrange(desc(type)) %>% 
  distinct(Node, .keep_all = T) %>%
  column_to_rownames("Node")

annotation_rows <- data.frame(networkID = rownames(pheatmap_input1_COSMOS))%>%
  left_join(distinct(nodes, networkID, TPP_set, PKN), by = c("networkID"))  %>%
  column_to_rownames("networkID")

annotation_colors = list(
  type = values_types,
  TPP_set = values_TPPset)
```

```{r, fig.width=7, fig.height=4}
my_palette <- c(colorRampPalette(colors = c("#67001F"))(n = 1),
              "gray91",
              c(colorRampPalette(colors = c("darkgreen"))(n = 1)))

pheatmap_input2_COSMOS <- pheatmap_input2_COSMOS[, abs(colSums(pheatmap_input2_COSMOS)) > 1]

p_heatmap_nodes2 <- pheatmap::pheatmap(pheatmap_input2_COSMOS, 
                   show_colnames = F, 
                   annotation_col = annotation_cols,
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   legend_breaks = c(-0.8, 0, 0.8), 
                   legend_labels = c("inactive", "missed", "active"), 
                   fontsize = 8,
                   main = "Comparison of network nodes of different TPP sets")

```

#### Paper

```{r, fig.width=7, fig.height=4}
my_palette <- c(colorRampPalette(colors = c("#67001F"))(n = 1),
              "gray91",
              c(colorRampPalette(colors = c("darkgreen"))(n = 1)))

pheatmap_input2_COSMOS <- pheatmap_input2_COSMOS[, abs(colSums(pheatmap_input2_COSMOS)) > 1]

test <- rownames(pheatmap_input2_COSMOS)
test <- gsub("monotony", "large", test)
test <- gsub("Fstatistic", "medium", test)
test <- gsub("highconfidence", "small", test)
test <- gsub("GSK", "PKN1", test)
test <- gsub("COSMOSnew", "PKN2", test)
test -> rownames(pheatmap_input2_COSMOS)

annotation_rows <- data.frame(id = test) %>% 
  separate(id, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F) %>% 
  dplyr::select(-replicate) %>% 
  column_to_rownames("id")

annotation_colors = list(
  type = values_types,
  TPP_set = values_TPPset_paper,
  PKN = values_PKN_paper)

p_heatmap_nodes2 <- pheatmap::pheatmap(pheatmap_input2_COSMOS, 
                   show_colnames = F, show_rownames = F,
                   annotation_col = annotation_cols,
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   legend_breaks = c(-0.8, 0, 0.8), 
                   legend_labels = c("inactive", "missed", "active"), 
                   fontsize = 10)

```


Visualize heatmap clustering as PCA.

```{r,  fig.width=7, fig.height=6}
x = prcomp(pheatmap_input2_COSMOS)
PCA_nodes_res = x$x %>% as.data.frame() %>% 
  rownames_to_column("networkID") %>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F)

p_pca_nodes <- ggplot(PCA_nodes_res, aes(x = PC1, y = PC2, colour = TPP_set, shape = PKN)) +
  geom_point(size = 5) +
  scale_colour_manual(values = values_TPPset) +
  theme_bw(14) +
  labs(title = "PCA of networks based on node activity")
p_pca_nodes
```


## Analysis of Edges

Extract edges and related node type information.

```{r}
TPP_nodes <- nodes %>% 
  filter(type == "TPP") %>% 
  dplyr::select(networkID, node, type)

edges <- map_dfr(result_list, 'combined_network', .id = "networkID")  %>%
  mutate(edge_combined = paste(from, sign, to, sep = "_")) %>%
  distinct(networkID, edge_combined, .keep_all = T)%>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F) %>% 
  left_join(TPP_nodes, by=c("networkID", "from" = "node"), suffix = c("", "_source")) %>% 
  left_join(TPP_nodes, by=c("networkID", "to" = "node"), suffix = c("_source", "_target"))
```

Summarise edge information

```{r}
TPP_edges <- edges %>% 
  filter(type_source == "TPP" | type_target == "TPP")%>% 
  group_by(PKN, TPP_set, replicate) %>% 
  summarise(count = n())

edge_summary <- edges %>% 
  group_by(PKN, TPP_set, replicate) %>% 
  summarise(count = n())

edge_summary <- bind_rows("all edges" = edge_summary, "TPP edges" = TPP_edges, .id = "edge_type") %>%
  group_by(edge_type, PKN, TPP_set) %>%
  summarise(mean = mean(count), sd = sd(count)) %>% 
  bind_rows(data.frame(edge_type = c("TPP edges", "TPP edges"),
                       PKN = c("GSK", "COSMOSnew"),
                       TPP_set = c("noTPP", "noTPP"),
                       mean = c(0,0), sd = c(NA, NA)))

```

### Number of edges

```{r, fig.width=7, fig.height=6}
p_edge_summary <- edge_summary %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = PKN,
             colour = PKN)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = PKN), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line() +
  scale_colour_manual(values = values_PKN) +
  scale_y_continuous(limits = c(0, 200)) +
  theme_bw(base_size = 14) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank(),
        legend.position = "top") +
  facet_wrap(~edge_type) +
  labs(x = "", y = "number of edges", title = "Number of edges per network")
p_edge_summary
```

Thesis plot

```{r, eval = F}
p_edge_summary <- edge_summary %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             group = PKN,
             colour = PKN)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd, group = PKN), width=.4, size = 1) +
  geom_point(size = 3) +
  geom_line() +
  scale_colour_manual(values = values_PKN) +
   scale_y_continuous(limits = c(0, 220)) +
  theme_bw(base_size = 16) +
  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust =1), panel.grid.minor = element_blank()) +
  facet_wrap(~edge_type) +
  #labs(x = "network", y = "total number of edges", title = "Number of edges per network") 
  labs(x = "", y = "number of edges")
p_edge_summary
```

### Heatmap

Clustering analysis of networks based on edege sign

```{r}
pheatmap_input_edges <- edges %>%
  mutate(sign = as.numeric(sign)) %>%
  acast(networkID ~ edge_combined, value.var = "sign", fill = 0)

# annotation_cols <- data.frame(edge = colnames(pheatmap_input_edges))%>%
#   left_join(distinct(edges, edge_combined, sign), by = c("edge"= "edge_combined")) %>%
#   column_to_rownames("edge")

annotation_rows <- data.frame(networkID = rownames(pheatmap_input_edges))%>%
  left_join(distinct(nodes, networkID, TPP_set, PKN), by = c("networkID"))  %>%
  column_to_rownames("networkID")
```


```{r, fig.width=7, fig.height=4}
my_palette <- c(colorRampPalette(colors = c("#67001F"))(n = 1),
              "gray91",
              c(colorRampPalette(colors = c("darkgreen"))(n = 1)))
annotation_colors = list(
  #type = c(Kinase="red", TF="yellow", other = "blue", TPP = "green"),
  TPP_set = values_TPPset,
  PKN = values_PKN)

pheatmap::pheatmap(pheatmap_input_edges, 
                   show_colnames = F, 
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   legend_breaks = c(-0.8, 0, 0.8), 
                   legend_labels = c("inhibition", "missed", "activation"),
                   fontsize = 8,
                   main = "Comparison of network edge sign\n of different TPP sets")

```

PCA analysis of edge clustering

```{r, fig.width=7, fig.height=6}
x = prcomp(pheatmap_input_edges)
PCA_edges_res = x$x %>% as.data.frame() %>% 
  rownames_to_column("networkID") %>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F)

p_pca_edges <- ggplot(PCA_edges_res, aes(x = PC1, y = PC2, colour = TPP_set, shape = PKN)) +
  geom_point(size = 5) +
  scale_colour_manual(values = values_TPPset) +
  theme_bw(14)+
  labs(title = "PCA of networks based on edge sign")
p_pca_edges
```

## Network level analysis

### Edge density

```{r, fig.width = 7, fig.height=6}
densities <- map(network_list, 'network_density', .id = "networkID") %>% 
  unlist()

densities <- data.frame(networkID = names(densities), density = unname(densities))%>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F)%>% 
  group_by(PKN, TPP_set) %>% 
  summarise(mean = mean(density), sd =sd(density))
 
p_densities <- densities %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             colour = PKN, group =PKN)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.4, size = 1) +
  geom_point(size = 4) +
  geom_line(size =1) +
  scale_colour_manual(values = values_PKN) +
  theme_bw(14) +
  theme(panel.grid.minor = element_blank(), axis.text.x = element_text(angle =45, hjust =1, vjust =1),
        legend.position = "right") +
  labs(x = "", y = "density", title = "Edge densities of different networks")
p_densities
```

Thesis plots

```{r, eval = F}
p_densities <- densities %>% 
  ggplot(aes(x = factor(TPP_set, levels = levels_TPPset), 
             y = mean,
             colour = PKN, group =PKN)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.4, size = 1) +
  geom_point(size = 4) +
  geom_line(size =1) +
  scale_colour_manual(values = values_PKN) +
  theme_bw(16) +
  theme(panel.grid.minor = element_blank(), axis.text.x = element_text(angle =45, hjust =1, vjust =1)) +
  #labs(x = "network", y = "total number of edges", title = "Number of edges per network") 
  labs(x = "", y = "density")
p_densities
```

### Network distance

```{r}
adjacency_matrices <- map(network_list, 'adj_matrix_m', .id = "networkID")
node_universe <- unique(unlist(unname(lapply(adjacency_matrices, colnames))))

mat_assim <- function(x){ 
  missing_cols = setdiff(node_universe, colnames(x))
  col_mat = matrix(nrow = nrow(x), ncol = length(missing_cols))
  x_col = cbind(x, col_mat)
  missing_rows <- setdiff(node_universe, rownames(x))
  row_mat = matrix(nrow = length(missing_rows), ncol = ncol(x_col))
  x_new = rbind(x_col, row_mat)
  colnames(x_new) <- c(colnames(x), setdiff(node_universe, colnames(x)))
  rownames(x_new) <- c(rownames(x), setdiff(node_universe, rownames(x)))
  x_new[is.na(x_new)] <- 0
  return(x_new)
  }

list2 <- lapply(adjacency_matrices, function(x) {
x = mat_assim(x)
})

network_hamm <- nd.hamming(A = list2)
distance_matrix <- as.matrix(network_hamm$D)
colnames(distance_matrix) <- names(list2)
rownames(distance_matrix) <- names(list2)

distance_long <- distance_matrix %>% 
  melt()
```


```{r, fig.width=7, fig.height=4}
my_palette <- c(colorRampPalette(c("darkslategrey", "darkcyan", "paleturquoise3", "white"))(n = 30))

p_networkheatmap <- pheatmap::pheatmap(distance_matrix,
                   show_colnames = F, 
                   annotation_colors = annotation_colors,
                   treeheight_col = 0, 
                   annotation_row = annotation_rows, 
                   color = my_palette,
                   fontsize = 8,
                   main = "Comparison of network edges\nof different TPP sets")
```


```{r, fig.width =7, fig.height=7}
hc_input <- as.dist(distance_matrix, diag = TRUE)
hc <- hclust(hc_input)

colours <- ifelse(grepl("GSK", hc$labels), "goldenrod", "mediumvioletred")

dendroNetwork(hc, width = 700, height = 700, textColour = colours, nodeColour = "darkgray", nodeStroke = 2)
```


<!-- ```{r} -->
<!-- MDS_netdist = cmdscale(distance_matrix) %>% -->
<!--   as.data.frame() %>%  -->
<!--   rownames_to_column("networkID") %>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F) -->

<!-- ggplot(MDS_netdist, aes(x = V1, y = V2, colour = TPP_set, shape = PKN)) + -->
<!--   geom_point(size = 3) + -->
<!--   scale_colour_manual(values = values_TPPset) + -->
<!--   theme_bw(16) -->
<!-- ``` -->

## Pathway check

```{r, message=F, warning=F, eval = FALSE}
# load any PKN as background, I chose the GSK PKN because bigger
load("results/2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1.RData")
pathways_universe <- COSMOS_object_fw1$meta_network %>% 
  mutate(source = gsub("X", "", source), target = gsub("X", "", target))
pathways_universe <- data.frame(protein = mapIds(org.Hs.eg.db, unique(c(pathways_universe$source, pathways_universe$target)),
                                                 'SYMBOL', 'ENTREZID'), NES = 1)
# get gathered pathway input
pathway_input_gathered <- map_dfr(result_list, 'nodes', .id = "networkID") %>% 
  dplyr::select(networkID, id)

# metabase enrichment
pathway_enrichment <- pathway_input_gathered %>% 
  group_by(networkID) %>% 
  mutate(enrichment = list(ORA_metabase(data.frame(protein = id, NES = 1), 
                                   pathways_universe, 
                                   "snetws")))
pathway_enrichment_unnest <- pathway_enrichment %>% 
  distinct(networkID, enrichment) %>% 
  unnest() %>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F)
pathway_enrichment_significant <- pathway_enrichment_unnest %>% 
  filter(qvalue < 0.05)
```

```{r, message=F, warning=F}
# load any PKN as background, I chose the GSK PKN because bigger
load("results/2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1.RData")
pathways_universe <- COSMOS_object_fw1$meta_network %>% 
  mutate(source = gsub("X", "", source), target = gsub("X", "", target))
pathways_universe <- data.frame(protein = mapIds(org.Hs.eg.db, unique(c(pathways_universe$source, pathways_universe$target)),
                                                 'SYMBOL', 'ENTREZID'), NES = 1)
# get gathered pathway input
pathway_input_gathered <- map_dfr(result_list, 'nodes', .id = "networkID") %>% 
  dplyr::select(networkID, id)

gene_lists <- pathway_input_gathered %>% 
  group_by(networkID) %>% 
  summarise(length = n())

# metabase enrichment
pathway_enrichment <- pathway_input_gathered %>% 
  group_by(networkID) %>% 
  mutate(enrichment = list(as.data.frame(GSE_RPA(geneList = id, universe = pathways_universe$protein))))

pathway_enrichment_unnest <- pathway_enrichment %>% 
  distinct(networkID, enrichment) %>% 
  unnest() %>% 
  separate(networkID, into = c("TPP_set", "replicate", "PKN"), sep = "_", remove = F) %>% 
  left_join(gene_lists, by = "networkID")
```


```{r, message=F, warning=F}
pathway_enrichment_significant <- pathway_enrichment_unnest %>% 
  filter(qvalue < 0.05) %>% 
  mutate(pathway = Description, Zscore = -log10(qvalue),
         pathway_size = as.numeric(gsub("/.+$", "", BgRatio)),
         Gene_Ratio = Count/pathway_size,
         LR = log2((Count/length)/(pathway_size/length(pathways_universe$protein))))
```


Disentangle the results

```{r}
pathways_perstrategy <- pathway_enrichment_significant%>% 
  filter(TPP_set != "noTPP") %>% 
  ungroup() %>% 
  distinct(PKN, TPP_set, pathway)%>% 
  group_by(PKN, pathway) %>% 
  mutate(number_TPPsets = n()) %>% 
  right_join(pathway_enrichment_significant, by = c("PKN", "TPP_set", "pathway"))

pathways_TPPunique <- pathways_perstrategy %>% 
   anti_join(filter(pathway_enrichment_significant, TPP_set == "noTPP" ), by = c("PKN","pathway"))

filter(pathway_enrichment_significant, TPP_set == "noTPP" )
```

TPP specific results

```{r}
x=pathways_TPPunique %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID)
knitr::kable(pathways_TPPunique %>% distinct(PKN, pathway, .keep_all = T)) %>% 
               #dplyr::select( number_TPPsets, Zscore, qvalue, r, intersect)) %>%
  kable_styling("striped", full_width = F)%>%
  scroll_box(width = "700px", height = "200px")
```

```{r, fig.width = 7, fig.height = 7}
myPalette <- colorRampPalette(c("darkslategrey", "darkcyan", "paleturquoise3"))
p_pathway_enrichment <- pathways_TPPunique %>% 
  distinct(PKN, pathway, .keep_all = T) %>% 
  group_by(PKN) %>% 
  top_n(10, wt = Zscore) %>% 
  ggplot(aes(x= Zscore,
             y = pathway,
             colour = qvalue,
             size = Count)) +
  geom_point() +
  scale_colour_gradientn(colours = myPalette(100)) +
  theme_bw(14) +
  theme(panel.grid.minor = element_blank(), legend.position = "top", axis.text.y = element_text(size = 9)) +
  facet_wrap(~PKN, ncol =1, scales = "free_y", strip.position = "right") +
  labs(y = "", x = "Z-score of pathway enrichment") +
  guides(size=guide_legend(ncol=2, title.position = "top"), colour = guide_legend(ncol = 2, direction = "vertical")) +
  labs(title = "TPP specific pathways")
p_pathway_enrichment
```


#### Paper

```{r, fig.width=7, fig.height=4}

group_cc <- "cell cycle|Cell Cycle|NIMA|APC|G2/|G1/|Mitotic|mitotic|Prometaphase|G2 Phase|G1 Phase|Prophase|cyclin|Cyclin"
group_DNA <- "Checkpoint|Repair|Damage|repair|damage| Strand Breaks|DNA double-strand break|Homologous Recombination|Chk1/Chk2|DNA Double Strand Break"

x_cc=pathways_TPPunique %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(grepl(group_cc, pathway))

x_DNA=pathways_TPPunique %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(grepl(group_DNA, pathway))

other <- pathways_TPPunique %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(!(pathway %in% c(x_cc$pathway, x_DNA$pathway)))

comparison_TPP <- bind_rows(
  "Cell \n Cycle" = x_cc, "DNA damage \n and repair" = x_DNA, "other" = other, .id = "pathway_family")


nonTPP <- filter(pathway_enrichment_significant, TPP_set == "noTPP" )

x_cc=nonTPP %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(grepl(group_cc, pathway))

x_DNA=nonTPP %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(grepl(group_DNA, pathway))

other <- nonTPP %>% ungroup() %>% distinct(PKN, pathway, .keep_all = T) %>% select(PKN,pathway,qvalue, Count, geneID, Gene_Ratio, pathway_size, LR) %>% 
  filter(!(pathway %in% c(x_cc$pathway, x_DNA$pathway)))

comparison_noTPP <- bind_rows(
  "Cell \n Cycle" = x_cc, "DNA damage \n and repair" = x_DNA, "other" = other, .id = "pathway_family") 

comparison = bind_rows("TPP" = comparison_TPP, "noTPP" = comparison_noTPP, .id = "network") %>% 
  filter(Count > 3 & pathway_size > 10) %>% 
  ungroup()%>% 
  filter(pathway_family != "other") 

p <- ggplot(comparison, aes(x = pathway_family, y = LR, colour = network)) +
  geom_boxplot(lwd =1) +
  stat_compare_means(aes(group = network), label.y = 5.5, label = "p.signif") +
  scale_colour_manual(values = c("TPP" = "mediumseagreen", "noTPP" = "grey50")) +
  cowplot::theme_cowplot(font_size = 16) +
  labs(x = "", y = "Log odds-ratio \n pathway enrichment") +
  theme(legend.position = "bottom")

p

```

```{r}
p = comparison %>% ggplot(aes(x = Gene_Ratio*100, y = -log10(qvalue), colour = network)) +
  geom_point(size=3) +
  theme_bw(14)
```



```{r, eval = F}
myPalette <- colorRampPalette(c("darkslategrey", "darkcyan", "paleturquoise3"))
p_pathway_enrichment <- pathways_TPPunique %>% 
  distinct(PKN, pathway, .keep_all = T) %>% 
  ggplot(aes(x= Zscore,
             y = reorder(pathway, -Zscore),
             colour = qvalue,
             size = proportion)) +
  geom_point() +
  scale_colour_gradientn(colours = myPalette(100)) +
  theme_bw(14) +
  theme(panel.grid.minor = element_blank()) +
  facet_wrap(~PKN, ncol =1, scales = "free_y", strip.position = "right") +
  labs(y = "", x = "Z-score of pathway enrichment")
p_pathway_enrichment
```


### DNA damage core

```{r, fig.width = 7, fig.height = 7}
# interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "DNA damage_Core"])

interesting_pathwaygenes <- c("ATR", "CHEK2, CNNA2", "FOXM1", "ATM", "PARP1", "PTTG1", "BRCA1", "CDC25")


pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges) %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visLegend() %>%
  visNodes(font = '30px arial black bold') %>% 
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE, manipulation = T)%>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 2, physics = F)
test

```

### Cell cycle

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "Cell cycle_Core"])

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '30px arial black bold') %>% 
    visLegend() %>% 
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)%>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 1, physics = F) 
test

```

### Development_Ossification and bone remodeling

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "Development_Ossification and bone remodeling"])

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred")) %>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visNodes(font = '28px arial black bold') %>% 
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE)%>%
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 1) 
test

```

### Apoptosis

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "Apoptosis_Apoptotic nucleus"])

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE, manipulation = T)%>%
    visNodes(font = '30px arial black bold') %>% 
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 2) 
test

```

### WNT/TGFbeta signaling

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "Signal transduction_WNT signaling"])

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE, manipulation = T)%>%
    visNodes(font = '30px arial black bold') %>% 
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 3) 
test

```

### Hippo signaling

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- c("WWTR1", "TEAD3", "TEAD4", "CREBBP", "ZEB2", "ETS1")

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type) %>% 
  distinct(id, .keep_all = T)

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE, manipulation = T)%>%
    visOptions( manipulation = T)%>%
    visNodes(font = '30px arial black bold') %>% 
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 3) 
test

```

### DNA damage checkpoint

```{r, fig.width = 7, fig.height = 7}
interesting_pathwaygenes <- c("ATM", "ATR", "CHEK2", "FOXM1", "CDC25B", "PARP1", "PTTG1", "BRCA1")
# simplify visualisation
exclusion_list <- c("CSF2", "PSMA6", "PSMA1", "PSMA2", "PSMB6", "MEN1", "KDM1A", "NFIC", "STAT2", "PRKAA1", "CDK5", "FZR1", "UBB", "UBC", "SOX2", "AKT1", "AKT2")

pathway_edges <- edges %>% 
  filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) %>% 
  filter(!(from %in% exclusion_list | to %in% exclusion_list)) %>% 
  filter(TPP_set != "noTPP") %>% 
  mutate(color = ifelse(PKN == "GSK", "#ffb90f", "mediumvioletred"))%>% 
  group_by(PKN, edge_combined) %>% 
  mutate(width = n()) %>% 
  ungroup() %>% 
  distinct(PKN, edge_combined, .keep_all = T)

pathway_nodes <- nodes %>% 
  filter(node %in% pathway_edges$from | node %in%  pathway_edges$to)%>% 
  mutate(label = node, id = node,
         measured = ifelse(is.na(measured), "active", measured),
         group = type,
         shape = ifelse(model == "PKN node", "triangle", "square")) %>% 
  distinct(id, .keep_all = T) %>% 
  dplyr::filter(!(node %in% exclusion_list))

test = visNetwork(pathway_nodes , pathway_edges,  height = "1000px", width = "100%") %>%
    visGroups(groupname = "TPP", color = "mediumseagreen") %>%
    visGroups(groupname = "TF", color = "#8b0a50") %>%
    visGroups(groupname = "Kinase", color = "darkcyan") %>%
    visGroups(groupname = "other", color = "gray") %>%
    # standard options
    visEdges(arrows = "to") %>%
    visLegend() %>%
    visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE, manipulation = T)%>%
   visOptions( manipulation = T)%>%
    visNodes(font = '40px arial black bold') %>% 
    visNetwork::visIgraphLayout(layout =  "layout_with_fr", randomSeed = 3) 
test



```

# GGrpah

```{r}
library(reactome.db)
pathwaymembers <- toTable(reactomePATHID2EXTID) %>%
  mutate(gene_name = mapIds(org.Hs.eg.db, gene_id, "SYMBOL", "ENTREZID"))

pathway_names <- toTable(reactomePATHID2NAME) %>%
  filter(grepl("Homo sapiens", path_name))

DNA_DR <- pathway_names %>%
  filter(grepl("DNA Damage|DNA damage|DNA Repair|DNA repair|Strand Breaks|DNA double-strand break|Homologous Recombination|DNA Double Strand Break", path_name))

cell_cycle <- pathway_names %>%
  filter(grepl("Cell cycle|cell cycle|APC|cyclin|Cyclin", path_name))


DNA_DR_members <- pathwaymembers %>%
  filter(DB_ID %in% DNA_DR$DB_ID) %>%
  distinct(gene_name)

cell_cycle_members <- pathwaymembers %>%
  filter(DB_ID %in% cell_cycle$DB_ID) %>%
  distinct(gene_name)
```


```{r, fig.width=7, fig.height=4}
library(ggraph)
graph <- edges %>%
  filter(TPP_set == "Fstatistic") %>% 
  as.data.frame() %>% 
  select(from, to, sign) %>% 
  distinct(from, to, sign) %>% 
  mutate(process = ifelse(from %in% cell_cycle_members$gene_name| to %in% cell_cycle_members$gene_name, "Cell Cycle", "other"),
         process = ifelse(from %in% DNA_DR_members$gene_name| to %in% DNA_DR_members$gene_name, "DNA damage and repair", process)) %>% 
  graph_from_data_frame()

set.seed(1)
p <- ggraph(graph, layout = "stress") +
  geom_edge_link(aes(colour = process)) +
  geom_node_point() +
  scale_edge_color_manual(values = c("Cell Cycle" = "palegreen3", "DNA damage and repair" = "darkgreen", "other" = "darkgrey"))+
  theme_graph() +
  theme(legend.position = "bottom") 
p
```



<!-- Depreciated -->

<!-- # Comparison COSMOS -->

<!-- ## Nodes -->

<!-- ```{r} -->
<!-- result_list <- mget(ls(pattern= "^res_.*COSMOS")) -->
<!-- names(result_list) <- c("noTPP_rep1", "noTPP_rep2", "noTPP_rep3", "Fstat_rep1", "Fstat_rep2", "Fstat_rep3", "highconf_rep1", "mono_rep1", "mono_rep2", "mono_rep3")  -->

<!-- nodes <- map_dfr(result_list, 'nodes', .id = "networkID") %>%  -->
<!--   dplyr::select(networkID, label, measured)  -->

<!-- network_list <- mget(ls(pattern= "^netprop_.*COSMOSnew")) -->
<!-- names(network_list) <-c("noTPP_rep1", "noTPP_rep2", "noTPP_rep3", "Fstat_rep1", "Fstat_rep2", "Fstat_rep3", "highconf_rep1", "mono_rep1", "mono_rep2", "mono_rep3")   -->


<!-- nodes <- map_dfr(network_list, 'overall_node_summary', .id = "networkID") %>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) %>%  -->
<!--   left_join(nodes, by = c("networkID", "node" = "label")) -->

<!-- nodes_summary <- nodes %>%  -->
<!--   filter(model != "missed") %>%  -->
<!--   group_by(TPP_set, replicate, type) %>%  -->
<!--   summarize(count = n()) %>%  -->
<!--   group_by(TPP_set, type) %>%  -->
<!--   mutate(mean = mean(count), sd =sd(count)) -->
<!-- ``` -->

<!-- #### Barplots -->

<!-- ```{r} -->
<!-- p_barplot_nodes <- nodes %>%  -->
<!--   #filter(model != "missed") %>%  -->
<!--   ggplot(aes(x = networkID, fill = TPP_set, group = TPP_set)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_TPPset) + -->
<!--   facet_wrap(~ model) + -->
<!--   theme_bw(12)+ -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + -->
<!--   labs(x = "network", y = "total number of nodes", title = "Number of nodes per network") -->
<!-- p_barplot_nodes -->

<!-- p_barplot_nodes_sum <- nodes_summary %>%  -->
<!--   ggplot(aes(x = TPP_set, y =mean, fill = type)) + -->
<!--   geom_bar(stat = "identity",position = position_dodge()) + -->
<!--   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2, -->
<!--                  position=position_dodge()) + -->
<!--   scale_fill_manual(values = values_types) + -->
<!--   facet_wrap(~ type, ncol = 4) + -->
<!--   theme_bw(12) + -->
<!--   labs(x = "TPP set", y = "mean number of nodes", title = "Mean number of nodes per node type and TPP set") -->
<!-- p_barplot_nodes_sum -->

<!-- p_TPP_nodes <- nodes %>%  -->
<!--   filter(type == "TPP") %>%  -->
<!--   ggplot(aes(x = replicate, fill = model, group = model)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_model) + -->
<!--   facet_wrap(~ TPP_set) + -->
<!--   theme_bw(12)+ -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+ -->
<!--   labs(x = "network", y = "number of TPP proteins", title = "Number of integrated and missed TPP proteins") -->
<!-- p_TPP_nodes -->
<!-- ``` -->

<!-- #### Centralities -->

<!-- ```{r} -->
<!-- centralities <- map_dfr(network_list, 'node_degrees', .id = "networkID") %>% -->
<!--   left_join(nodes, by = c("Node"= "node", "networkID")) %>%  -->
<!--   group_by(PKN, TPP_set, Node) %>%  -->
<!--   summarise(mean = mean(degree), sd= sd(degree)) -->

<!-- p_centralities <- ggplot(centralities, aes(x = TPP_set, y = mean, colour = type, fill = type)) + -->
<!--   geom_violin(alpha = 0.4) + -->
<!--   geom_point(position = position_jitterdodge(0.5)) + -->
<!--   geom_text_repel(aes(label = Node), size = 2) + -->
<!--   scale_fill_manual(values = values_types) + -->
<!--   scale_colour_manual(values = values_types) + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1)) + -->
<!--   facet_wrap(~type, ncol = 4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(x = "TPP set", y = "node degree", title = "Degree centrality per TPP set and protein type") -->
<!-- p_centralities -->
<!-- ``` -->

<!-- ### Heatmap -->

<!-- ```{r} -->
<!-- pheatmap_input1_COSMOS <- nodes %>% -->
<!--   filter(model != "missed") %>%  -->
<!--   mutate(conv = ifelse(model == "additional", -1, 0), -->
<!--          conv = ifelse(model == "input", 1, conv)) %>%  -->
<!--   distinct(networkID, node, .keep_all = T)%>% -->
<!--   acast(networkID ~ node, value.var = "conv", fill = 0) -->

<!-- pheatmap_input2_COSMOS <- nodes %>% -->
<!--   filter(model != "missed") %>%  -->
<!--   mutate(conv = ifelse(measured == "inactive", -1, 0), -->
<!--          conv = ifelse(measured == "active", 1, conv)) %>%  -->
<!--   distinct(networkID, node, .keep_all = T)%>% -->
<!--   acast(networkID ~ node, value.var = "conv", fill = 0) -->



<!-- # check <- pheatmap_input %>%  -->
<!-- #   dplyr::select(networkID, node) -->
<!-- #  -->
<!-- # check[duplicated(check[,1:2]),] -->

<!-- annotation_cols <- data.frame(Node = colnames(pheatmap_input1_COSMOS))%>% -->
<!--   left_join(distinct(nodes, node, type), by = c("Node"= "node")) %>%  -->
<!--   arrange(desc(type)) %>%  -->
<!--   distinct(Node, .keep_all = T) %>% -->
<!--   column_to_rownames("Node") -->

<!-- annotation_rows <- data.frame(networkID = rownames(pheatmap_input1_COSMOS))%>% -->
<!--   left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>% -->
<!--   column_to_rownames("networkID") -->


<!-- my_palette <- c(colorRampPalette(colors = c("darkolivegreen4"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("darkorange1"))(n = 1))) -->

<!-- annotation_colors = list( -->
<!--   type = values_types, -->
<!--   TPP_set = values_TPPset) -->

<!-- p_heatmap_nodes1 <- pheatmap::pheatmap(pheatmap_input1_COSMOS,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_col = annotation_cols, -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("additional", "missed", "input"),  -->
<!--                    main = "Comparison of network nodes of different TPP sets") -->

<!-- my_palette <- c(colorRampPalette(colors = c("darkred"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("navy"))(n = 1))) -->


<!-- p_heatmap_nodes2 <- pheatmap::pheatmap(pheatmap_input2_COSMOS,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_col = annotation_cols, -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("inactive", "missed", "active"),  -->
<!--                    main = "Comparison of network nodes of different TPP sets") -->


<!-- ``` -->

<!-- ```{r} -->


<!-- ``` -->


<!-- ## Edges -->

<!-- ```{r} -->
<!-- result_list <- mget(ls(pattern= "^res_.*COSMOS")) -->
<!-- names(result_list) <-  c("noTPP_rep1", "noTPP_rep2", "noTPP_rep3", "Fstat_rep1", "Fstat_rep2", "Fstat_rep3", "highconf_rep1", "mono_rep1", "mono_rep2", "mono_rep3") -->

<!-- TPP_nodes <- nodes %>%  -->
<!--   filter(type == "TPP") %>%  -->
<!--   dplyr::select(networkID, node, type) -->

<!-- edges <- map_dfr(result_list, 'combined_network', .id = "networkID")  %>% -->
<!--   mutate(edge_combined = paste(from, sign, to, sep = "_")) %>% -->
<!--   distinct(networkID, edge_combined, .keep_all = T)%>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) %>%  -->
<!--   left_join(TPP_nodes, by=c("networkID", "from" = "node"), suffix = c("", "_source")) %>%  -->
<!--   left_join(TPP_nodes, by=c("networkID", "to" = "node"), suffix = c("_source", "_target")) -->


<!-- ``` -->

<!-- ```{r} -->
<!-- p_barplot_edges <- edges %>%  -->
<!--   ggplot(aes(x = replicate,  fill = networkID)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   theme_bw(12)+ -->
<!--   facet_wrap(~TPP_set, ncol =4)+ -->
<!--   labs(x = "network", y = "total number of edges", title = "Number of edges per network") -->
<!-- p_barplot_edges -->

<!-- p_TPP_edges <- edges %>%  -->
<!--   filter(type_source == "TPP" | type_target == "TPP") %>%  -->
<!--   ggplot(aes(x = replicate,  fill = networkID)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_networkID) +   -->
<!--   theme_bw(12) + -->
<!--   facet_wrap(~TPP_set, ncol =4)+ -->
<!--   labs(x = "network", y = "number of edges from/to TPP proteins", title = "Number of TPP edges per network") -->
<!-- p_TPP_edges -->
<!-- ``` -->


<!-- ```{r} -->
<!-- pheatmap_input_edges <- edges %>% -->
<!--   mutate(sign = as.numeric(sign)) %>% -->
<!--   acast(networkID ~ edge_combined, value.var = "sign", fill = 0) -->

<!-- # annotation_cols <- data.frame(edge = colnames(pheatmap_input_edges))%>% -->
<!-- #   left_join(distinct(edges, edge_combined, sign), by = c("edge"= "edge_combined")) %>% -->
<!-- #   column_to_rownames("edge") -->

<!-- annotation_rows <- data.frame(networkID = rownames(pheatmap_input_edges))%>% -->
<!--   left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>% -->
<!--   column_to_rownames("networkID") -->


<!-- my_palette <- c(colorRampPalette(colors = c("darkred"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("navy"))(n = 1))) -->

<!-- annotation_colors = list( -->
<!--   #type = c(Kinase="red", TF="yellow", other = "blue", TPP = "green"), -->
<!--   TPP_set = values_TPPset) -->

<!-- pheatmap::pheatmap(pheatmap_input_edges,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("inhibition", "missed", "activation"), -->
<!--                    main = "Comparison of network edges of different TPP sets") -->

<!-- ``` -->

<!-- ```{r} -->
<!-- library(networkD3) -->
<!-- edge_binarized <- edges %>%  -->
<!--   acast(edge_combined~networkID, fun.aggregate = function(x){as.integer(length(x) > 0)}) -->

<!-- overlaps <- t(edge_binarized) %*% edge_binarized -->

<!-- p <- chordNetwork(Data = overlaps,  -->
<!--              width = 500,  -->
<!--              height = 500, -->
<!--              colourScale = c("darkcyan", "darkcyan","darkcyan", "mediumvioletred","mediumvioletred", "mediumvioletred","goldenrod", "goldenrod", "goldenrod", "darkolivegreen", "darkolivegreen", "darkolivegreen"), -->
<!--              labels = colnames(overlaps)) -->

<!-- p -->
<!-- ``` -->

<!-- ## Networks -->

<!-- ```{r} -->
<!-- densities <- map(network_list, 'network_density', .id = "networkID") %>%  -->
<!--   unlist() -->

<!-- densities <- data.frame(networkID = names(densities), density = unname(densities))%>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) -->

<!-- p_densities <- ggplot(densities, aes(x=replicate, y = density, fill = networkID)) + -->
<!--   geom_bar(stat = "identity") + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   facet_wrap(~TPP_set, ncol = 4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(x = "network", y = "density", title = "Comparison of network densities for different TPP sets") -->
<!-- p_densities -->
<!-- ``` -->


<!-- ```{r} -->
<!-- library(NetworkDistance) -->
<!-- library(Rcpp) -->
<!-- library(RColorBrewer) -->
<!-- adjacency_matrices <- map(network_list, 'adj_matrix_m', .id = "networkID") -->

<!-- node_universe <- unique(unlist(unname(lapply(adjacency_matrices, colnames)))) -->


<!-- mat_assim <- function(x){  -->
<!--   missing_cols = setdiff(node_universe, colnames(x)) -->
<!--   col_mat = matrix(nrow = nrow(x), ncol = length(missing_cols)) -->
<!--   x_col = cbind(x, col_mat) -->
<!--   missing_rows <- setdiff(node_universe, rownames(x)) -->
<!--   row_mat = matrix(nrow = length(missing_rows), ncol = ncol(x_col)) -->
<!--   x_new = rbind(x_col, row_mat) -->
<!--   colnames(x_new) <- c(colnames(x), setdiff(node_universe, colnames(x))) -->
<!--   rownames(x_new) <- c(rownames(x), setdiff(node_universe, rownames(x))) -->
<!--   x_new[is.na(x_new)] <- 0 -->
<!--   return(x_new) -->
<!--   } -->

<!-- list2 <- lapply(adjacency_matrices, function(x) { -->
<!-- x = mat_assim(x) -->
<!-- }) -->

<!-- network_hamm <- nd.hamming(A = list2) -->

<!-- distance_matrix <- as.matrix(network_hamm$D) -->
<!-- colnames(distance_matrix) <- names(list2) -->
<!-- rownames(distance_matrix) <- names(list2) -->

<!-- distance_long <- distance_matrix %>%  -->
<!--   melt() -->

<!-- myPalette <- colorRampPalette(c("darkslategrey", "darkcyan", "paleturquoise3")) -->
<!-- levels = c("highconf_rep1", "highconf_rep2","highconf_rep3", "mono_rep1", "mono_rep2", "mono_rep3", "Fstat_rep1","Fstat_rep2", "Fstat_rep3", "noTPP_rep1", "noTPP_rep2", "noTPP_rep3") -->

<!-- p_network_distances <- ggplot(distance_long, aes(x = factor(Var1, levels = levels), y = factor(Var2, levels = levels), fill = value)) + -->
<!--   geom_raster() + -->
<!--   scale_fill_gradientn(colours = myPalette(100)) + -->
<!--   theme_bw(12) + -->
<!--   labs(x = "", y = "", title = "Comparison of network hamming distances between different TPP sets") -->
<!-- p_network_distances -->

<!-- ``` -->


<!-- ```{r} -->
<!-- library(NetworkDistance) -->
<!-- library(networkD3) -->
<!-- # # A connection data frame is a list of flows with intensity for each flow -->
<!-- # links <- data.frame( -->
<!-- #   source=distance_long$Var1,  -->
<!-- #   target=distance_long$Var2,  -->
<!-- #   value=distance_long$value -->
<!-- #   ) %>%  -->
<!-- #   filter(source != target) %>%  -->
<!-- #   group_by(source, target) %>% -->
<!-- #   mutate(edge_id = paste(sort(unique(c(source, target))), collapse=" ")) %>%  -->
<!-- #   ungroup() %>%  -->
<!-- #   distinct(edge_id, .keep_all = T) -->
<!-- #   -->
<!-- # # From these flows we need to create a node data frame: it lists every entities involved in the flow -->
<!-- # nodes <- data.frame( -->
<!-- #   name=c(as.character(links$source),  -->
<!-- #   as.character(links$target)) %>% unique() -->
<!-- # ) -->
<!-- #   -->
<!-- # # With networkD3, connection must be provided using id, not using real name like in the links dataframe.. So we need to reformat it. -->
<!-- # links$IDsource <- match(links$source, nodes$name)-1  -->
<!-- # links$IDtarget <- match(links$target, nodes$name)-1 -->
<!-- #   -->
<!-- # # Make the Network -->
<!-- # p <- sankeyNetwork(Links = links, Nodes = nodes, -->
<!-- #               Source = "IDsource", Target = "IDtarget", -->
<!-- #               Value = "value", NodeID = "name",  -->
<!-- #               sinksRight=FALSE) -->

<!-- test <- as.dist(distance_matrix, diag = TRUE) -->
<!-- hc <- hclust(test) -->

<!-- dendroNetwork(hc, height = 500, nodeColour = "darkcyan") -->
<!-- ``` -->



<!-- ### Pathway check -->

<!-- ```{r} -->
<!-- ORA_metabase = function(viperRes, universe, GOs) { #, myGO -->

<!--   entities <-  filter(viperRes, NES != 0) %>% arrange(NES, decrease=T) -->

<!--   symbols_temp <- entities$protein -->
<!--   mapping_symbole_to_entrez_temp <- mapIds(org.Hs.eg.db, symbols_temp, 'ENTREZID', 'SYMBOL') -->
<!--   entities$entrezid <- mapping_symbole_to_entrez_temp[entities$protein] -->

<!--   gene_list <-  entities$entrezid -->
<!--   maps <-  load.ontology(GOs,  type = "gene") #load.ontology("maps", type = "gene") -->

<!--   entities <-  filter(universe, NES != 0) %>% arrange(NES, decrease=T) -->

<!--   symbols_temp <- entities$protein -->
<!--   mapping_symbole_to_entrez_temp <- mapIds(org.Hs.eg.db, symbols_temp, 'ENTREZID', 'SYMBOL') -->
<!--   entities$entrezid <- mapping_symbole_to_entrez_temp[entities$protein] -->

<!--   universe <-  entities$entrezid -->


<!--   er <- metabaser::enrichment(genelist = as.character(gene_list),  -->
<!--                               background.list = as.character(universe),  -->
<!--                               ontology = maps, -->
<!--                               min.r = 2, -->
<!--                               alpha = 1) -->


<!--   er_det <- er %>% mutate(pathway = rownames((er))) %>% -->
<!--     mutate(pathway_proteins = maps[pathway]) %>% -->
<!--     mutate(gene_list = list(gene_list)) %>% -->
<!--     rowwise() %>% -->
<!--     mutate(intersect =  list(intersect(gene_list, pathway_proteins))) %>% -->
<!--     rowwise() %>% -->
<!--     mutate(intersect = list( as.vector( mapIds(org.Hs.eg.db, intersect, 'SYMBOL', 'ENTREZID')))) %>% -->
<!--     mutate(proportion = r/n) %>% -->
<!--     dplyr::select(-pathway_proteins, -gene_list) -->
<!--   return(er_det) -->
<!-- } -->

<!-- ``` -->


<!-- ```{r, message=F, warning=F} -->
<!-- pathways_universe <- COSMOS_object_fw1$meta_network %>%  -->
<!--   mutate(source = gsub("X", "", source), target = gsub("X", "", target)) -->

<!-- pathways_universe <- data.frame(protein = mapIds(org.Hs.eg.db, unique(c(pathways_universe$source, pathways_universe$target)), -->
<!--                                                  'SYMBOL', 'ENTREZID'), NES = 1) -->


<!-- pathway_input_gathered <- map_dfr(result_list, 'nodes', .id = "networkID") %>%  -->
<!--   dplyr::select(networkID, id) -->


<!-- pathway_enrichment <- pathway_input_gathered %>%  -->
<!--   #nest(c(id))%>%  -->
<!--   group_by(networkID) %>%  -->
<!--   mutate(enrichment = list(ORA_metabase(data.frame(protein = id, NES = 1),  -->
<!--                                    pathways_universe,  -->
<!--                                    "snetws"))) -->


<!-- pathway_enrichment_unnest <- pathway_enrichment %>%  -->
<!--   distinct(networkID, enrichment) %>%  -->
<!--   unnest() %>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) -->


<!-- pathway_enrichment_significant <- pathway_enrichment_unnest %>%  -->
<!--   filter(qvalue < 0.05) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- p_pathway_enrichment <- ggplot(pathway_enrichment_significant, aes(y = Zscore, x = reorder(pathway, -Zscore), fill = networkID)) + -->
<!--   geom_bar(stat = "identity") + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   coord_flip() + -->
<!--   facet_wrap(~TPP_set, ncol =4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(y = "Zscore pathway enrichment", x = "pathway", title = "Significant pathways of network nodes") -->
<!-- p_pathway_enrichment -->
<!-- ``` -->

<!-- ```{r} -->
<!-- pathways_all_three <- pathway_enrichment_significant %>%  -->
<!--   filter(TPP_set != "noTPP") %>%  -->
<!--   ungroup() %>%  -->
<!--   distinct(TPP_set, pathway) %>%  -->
<!--   group_by(pathway) %>%  -->
<!--   mutate(number_TPPsets = n()) %>%  -->
<!--   right_join(pathway_enrichment_significant, by = c("TPP_set", "pathway")) %>%  -->
<!--   filter(number_TPPsets ==3) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- knitr::kable(pathways_all_three %>% distinct(pathway)) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- pathways_TPPunique <- pathways_all_three %>%  -->
<!--    anti_join(filter(pathway_enrichment_significant, TPP_set == "noTPP" ), by = "pathway") -->
<!-- ``` -->

<!-- ```{r} -->
<!-- knitr::kable(pathways_TPPunique %>% distinct(pathway)) -->
<!-- ``` -->


<!-- # Comparison GSK -->


<!-- ### Nodes -->

<!-- ```{r} -->
<!-- result_list <- mget(ls(pattern= "^res_.*GSK")) -->
<!-- # names(result_list) <- c("highconf_rep1","highconf_rep2","highconf_rep3","noTPP_rep1", "noTPP_rep2", "noTPP_rep3","mono_rep1", "mono_rep2", "mono_rep3", "Fstat_rep1", "Fstat_rep2", "Fstat_rep3")  -->
<!-- names(result_list) <- c("highconf_rep1","highconf_rep2", -->
<!--                         "Fstat_rep1", "Fstat_rep2", "Fstat_rep3", -->
<!--                         "highconf_rep3","mono_rep1", "mono_rep2", "mono_rep3", -->
<!--                         "noTPP_rep1", "noTPP_rep2", "noTPP_rep3")  -->

<!-- nodes <- map_dfr(result_list, 'nodes', .id = "networkID") %>%  -->
<!--   dplyr::select(networkID, label, measured)  -->

<!-- network_list <- mget(ls(pattern= "^netprop_.*GSK")) -->
<!-- # names(network_list) <-c("highconf_rep1","highconf_rep2","highconf_rep3","noTPP_rep1", "noTPP_rep2", "noTPP_rep3","mono_rep1", "mono_rep2", "mono_rep3","Fstat_rep1", "Fstat_rep2", "Fstat_rep3")  -->

<!-- names(network_list) <- c("highconf_rep1","highconf_rep2", -->
<!--                         "Fstat_rep1", "Fstat_rep2", "Fstat_rep3", -->
<!--                         "highconf_rep3","mono_rep1", "mono_rep2", "mono_rep3", -->
<!--                         "noTPP_rep1", "noTPP_rep2", "noTPP_rep3")  -->
<!-- nodes <- map_dfr(network_list, 'overall_node_summary', .id = "networkID") %>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) %>%  -->
<!--   left_join(nodes, by = c("networkID", "node" = "label")) -->

<!-- nodes_summary <- nodes %>%  -->
<!--   filter(model != "missed") %>%  -->
<!--   group_by(TPP_set, replicate, type) %>%  -->
<!--   summarize(count = n()) %>%  -->
<!--   group_by(TPP_set, type) %>%  -->
<!--   mutate(mean = mean(count), sd =sd(count)) -->
<!-- ``` -->

<!-- #### Barplots -->

<!-- ```{r} -->
<!-- p_barplot_nodes <- nodes %>%  -->
<!--   #filter(model != "missed") %>%  -->
<!--   ggplot(aes(x = networkID, fill = TPP_set, group = TPP_set)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_TPPset) + -->
<!--   facet_wrap(~ model) + -->
<!--   theme_bw(12)+ -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) + -->
<!--   labs(x = "network", y = "total number of nodes", title = "Number of nodes per network") -->
<!-- p_barplot_nodes -->

<!-- p_barplot_nodes_sum <- nodes_summary %>%  -->
<!--   ggplot(aes(x = TPP_set, y =mean, fill = type)) + -->
<!--   geom_bar(stat = "identity",position = position_dodge()) + -->
<!--   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2, -->
<!--                  position=position_dodge()) + -->
<!--   scale_fill_manual(values = values_types) + -->
<!--   facet_wrap(~ type, ncol = 4) + -->
<!--   theme_bw(12) + -->
<!--   labs(x = "TPP set", y = "mean number of nodes", title = "Mean number of nodes per node type and TPP set") -->
<!-- p_barplot_nodes_sum -->

<!-- p_TPP_nodes <- nodes %>%  -->
<!--   filter(type == "TPP") %>%  -->
<!--   ggplot(aes(x = replicate, fill = model, group = model)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_model) + -->
<!--   facet_wrap(~ TPP_set) + -->
<!--   theme_bw(12)+ -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))+ -->
<!--   labs(x = "network", y = "number of TPP proteins", title = "Number of integrated and missed TPP proteins") -->
<!-- p_TPP_nodes -->
<!-- ``` -->

<!-- #### Centralities -->

<!-- ```{r} -->
<!-- centralities <- map_dfr(network_list, 'node_degrees', .id = "networkID") %>% -->
<!--   left_join(nodes, by = c("Node"= "node", "networkID")) -->

<!-- p_centralities <- ggplot(centralities, aes(x = TPP_set, y = degree, colour = type, fill = type)) + -->
<!--   geom_violin(alpha = 0.4) + -->
<!--   geom_point(position = position_jitterdodge(0.5)) + -->
<!--   scale_fill_manual(values = values_types) + -->
<!--   scale_colour_manual(values = values_types) + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1)) + -->
<!--   facet_wrap(~type, ncol = 4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(x = "TPP set", y = "node degree", title = "Degree centrality per TPP set and protein type") -->
<!-- p_centralities -->
<!-- ``` -->

<!-- ### Heatmap -->

<!-- ```{r} -->
<!-- pheatmap_input1 <- nodes %>% -->
<!--   filter(model != "missed") %>%  -->
<!--   mutate(conv = ifelse(model == "additional", -1, 0), -->
<!--          conv = ifelse(model == "input", 1, conv)) %>%  -->
<!--   distinct(networkID, node, .keep_all = T)%>% -->
<!--   acast(networkID ~ node, value.var = "conv", fill = 0) -->

<!-- pheatmap_input2 <- nodes %>% -->
<!--   filter(model != "missed") %>%  -->
<!--   mutate(conv = ifelse(measured == "inactive", -1, 0), -->
<!--          conv = ifelse(measured == "active", 1, conv)) %>%  -->
<!--   distinct(networkID, node, .keep_all = T)%>% -->
<!--   acast(networkID ~ node, value.var = "conv", fill = 0) -->



<!-- # check <- pheatmap_input %>%  -->
<!-- #   dplyr::select(networkID, node) -->
<!-- #  -->
<!-- # check[duplicated(check[,1:2]),] -->

<!-- annotation_cols <- data.frame(Node = colnames(pheatmap_input1))%>% -->
<!--   left_join(distinct(nodes, node, type), by = c("Node"= "node")) %>%  -->
<!--   arrange(desc(type)) %>%  -->
<!--   distinct(Node, .keep_all = T) %>% -->
<!--   column_to_rownames("Node") -->

<!-- annotation_rows <- data.frame(networkID = rownames(pheatmap_input1))%>% -->
<!--   left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>% -->
<!--   column_to_rownames("networkID") -->


<!-- my_palette <- c(colorRampPalette(colors = c("darkolivegreen4"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("darkorange1"))(n = 1))) -->

<!-- annotation_colors = list( -->
<!--   type = values_types, -->
<!--   TPP_set = values_TPPset) -->

<!-- p_heatmap_nodes1 <- pheatmap::pheatmap(pheatmap_input1,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_col = annotation_cols, -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("additional", "missed", "input"),  -->
<!--                    main = "Comparison of network nodes of different TPP sets") -->

<!-- my_palette <- c(colorRampPalette(colors = c("darkred"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("navy"))(n = 1))) -->


<!-- p_heatmap_nodes2 <- pheatmap::pheatmap(pheatmap_input2,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_col = annotation_cols, -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("inactive", "missed", "active"),  -->
<!--                    main = "Comparison of network nodes of different TPP sets") -->


<!-- ``` -->

<!-- ```{r} -->


<!-- ``` -->


<!-- ### Edges -->

<!-- ```{r} -->

<!-- TPP_nodes <- nodes %>%  -->
<!--   filter(type == "TPP") %>%  -->
<!--   dplyr::select(networkID, node, type) -->

<!-- edges <- map_dfr(result_list, 'combined_network', .id = "networkID")  %>% -->
<!--   mutate(edge_combined = paste(from, sign, to, sep = "_")) %>% -->
<!--   distinct(networkID, edge_combined, .keep_all = T)%>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) %>%  -->
<!--   left_join(TPP_nodes, by=c("networkID", "from" = "node"), suffix = c("", "_source")) %>%  -->
<!--   left_join(TPP_nodes, by=c("networkID", "to" = "node"), suffix = c("_source", "_target")) -->


<!-- ``` -->

<!-- ```{r} -->
<!-- p_barplot_edges <- edges %>%  -->
<!--   ggplot(aes(x = replicate,  fill = networkID)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   theme_bw(12)+ -->
<!--   facet_wrap(~TPP_set, ncol =4)+ -->
<!--   labs(x = "network", y = "total number of edges", title = "Number of edges per network") -->
<!-- p_barplot_edges -->

<!-- p_TPP_edges <- edges %>%  -->
<!--   filter(type_source == "TPP" | type_target == "TPP") %>%  -->
<!--   ggplot(aes(x = replicate,  fill = networkID)) + -->
<!--   geom_bar(position = position_dodge()) + -->
<!--   scale_fill_manual(values = values_networkID) +   -->
<!--   theme_bw(12) + -->
<!--   facet_wrap(~TPP_set, ncol =4)+ -->
<!--   labs(x = "network", y = "number of edges from/to TPP proteins", title = "Number of TPP edges per network") -->
<!-- p_TPP_edges -->
<!-- ``` -->


<!-- ```{r} -->
<!-- pheatmap_input_edges <- edges %>% -->
<!--   mutate(sign = as.numeric(sign)) %>% -->
<!--   acast(networkID ~ edge_combined, value.var = "sign", fill = 0) -->

<!-- # annotation_cols <- data.frame(edge = colnames(pheatmap_input_edges))%>% -->
<!-- #   left_join(distinct(edges, edge_combined, sign), by = c("edge"= "edge_combined")) %>% -->
<!-- #   column_to_rownames("edge") -->

<!-- annotation_rows <- data.frame(networkID = rownames(pheatmap_input_edges))%>% -->
<!--   left_join(distinct(nodes, networkID, TPP_set), by = c("networkID"))  %>% -->
<!--   column_to_rownames("networkID") -->


<!-- my_palette <- c(colorRampPalette(colors = c("darkred"))(n = 1), -->
<!--               "gray91", -->
<!--               c(colorRampPalette(colors = c("navy"))(n = 1))) -->

<!-- annotation_colors = list( -->
<!--   #type = c(Kinase="red", TF="yellow", other = "blue", TPP = "green"), -->
<!--   TPP_set = values_TPPset) -->

<!-- pheatmap::pheatmap(pheatmap_input_edges,  -->
<!--                    show_colnames = F,  -->
<!--                    annotation_colors = annotation_colors, -->
<!--                    treeheight_col = 0,  -->
<!--                    annotation_row = annotation_rows,  -->
<!--                    color = my_palette, -->
<!--                    legend_breaks = c(-0.8, 0, 0.8),  -->
<!--                    legend_labels = c("inhibition", "missed", "activation"), -->
<!--                    main = "Comparison of network edges of different TPP sets") -->

<!-- ``` -->

<!-- ```{r} -->
<!-- library(networkD3) -->
<!-- edge_binarized <- edges %>%  -->
<!--   acast(edge_combined~networkID, fun.aggregate = function(x){as.integer(length(x) > 0)}) -->

<!-- overlaps <- t(edge_binarized) %*% edge_binarized -->

<!-- p <- chordNetwork(Data = overlaps,  -->
<!--              width = 500,  -->
<!--              height = 500, -->
<!--              colourScale = c("darkcyan", "darkcyan","darkcyan", "mediumvioletred","mediumvioletred", "mediumvioletred","goldenrod", "goldenrod", "goldenrod", "darkolivegreen", "darkolivegreen", "darkolivegreen"), -->
<!--              labels = colnames(overlaps)) -->

<!-- p -->
<!-- ``` -->

<!-- ### Networks -->

<!-- ```{r} -->
<!-- densities <- map(network_list, 'network_density', .id = "networkID") %>%  -->
<!--   unlist() -->

<!-- densities <- data.frame(networkID = names(densities), density = unname(densities))%>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) -->

<!-- p_densities <- ggplot(densities, aes(x=replicate, y = density, fill = networkID)) + -->
<!--   geom_bar(stat = "identity") + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   facet_wrap(~TPP_set, ncol = 4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(x = "network", y = "density", title = "Comparison of network densities for different TPP sets") -->
<!-- p_densities -->
<!-- ``` -->


<!-- ```{r} -->
<!-- library(NetworkDistance) -->
<!-- library(Rcpp) -->
<!-- library(RColorBrewer) -->
<!-- adjacency_matrices <- map(network_list, 'adj_matrix_m', .id = "networkID") -->

<!-- node_universe <- unique(unlist(unname(lapply(adjacency_matrices, colnames)))) -->


<!-- mat_assim <- function(x){  -->
<!--   missing_cols = setdiff(node_universe, colnames(x)) -->
<!--   col_mat = matrix(nrow = nrow(x), ncol = length(missing_cols)) -->
<!--   x_col = cbind(x, col_mat) -->
<!--   missing_rows <- setdiff(node_universe, rownames(x)) -->
<!--   row_mat = matrix(nrow = length(missing_rows), ncol = ncol(x_col)) -->
<!--   x_new = rbind(x_col, row_mat) -->
<!--   colnames(x_new) <- c(colnames(x), setdiff(node_universe, colnames(x))) -->
<!--   rownames(x_new) <- c(rownames(x), setdiff(node_universe, rownames(x))) -->
<!--   x_new[is.na(x_new)] <- 0 -->
<!--   return(x_new) -->
<!--   } -->

<!-- list2 <- lapply(adjacency_matrices, function(x) { -->
<!-- x = mat_assim(x) -->
<!-- }) -->

<!-- network_hamm <- nd.hamming(A = list2) -->

<!-- distance_matrix <- as.matrix(network_hamm$D) -->
<!-- colnames(distance_matrix) <- names(list2) -->
<!-- rownames(distance_matrix) <- names(list2) -->

<!-- distance_long <- distance_matrix %>%  -->
<!--   melt() -->

<!-- myPalette <- colorRampPalette(c("darkslategrey", "darkcyan", "paleturquoise3")) -->
<!-- levels = c("highconf_rep1", "highconf_rep2","highconf_rep3", "mono_rep1", "mono_rep2", "mono_rep3", "Fstat_rep1","Fstat_rep2", "Fstat_rep3", "noTPP_rep1", "noTPP_rep2", "noTPP_rep3") -->

<!-- p_network_distances <- ggplot(distance_long, aes(x = factor(Var1, levels = levels), y = factor(Var2, levels = levels), fill = value)) + -->
<!--   geom_raster() + -->
<!--   scale_fill_gradientn(colours = myPalette(100)) + -->
<!--   theme_bw(12) + -->
<!--   labs(x = "", y = "", title = "Comparison of network hamming distances between different TPP sets") -->
<!-- p_network_distances -->

<!-- ``` -->


<!-- ```{r} -->
<!-- library(NetworkDistance) -->
<!-- library(networkD3) -->
<!-- # # A connection data frame is a list of flows with intensity for each flow -->
<!-- # links <- data.frame( -->
<!-- #   source=distance_long$Var1,  -->
<!-- #   target=distance_long$Var2,  -->
<!-- #   value=distance_long$value -->
<!-- #   ) %>%  -->
<!-- #   filter(source != target) %>%  -->
<!-- #   group_by(source, target) %>% -->
<!-- #   mutate(edge_id = paste(sort(unique(c(source, target))), collapse=" ")) %>%  -->
<!-- #   ungroup() %>%  -->
<!-- #   distinct(edge_id, .keep_all = T) -->
<!-- #   -->
<!-- # # From these flows we need to create a node data frame: it lists every entities involved in the flow -->
<!-- # nodes <- data.frame( -->
<!-- #   name=c(as.character(links$source),  -->
<!-- #   as.character(links$target)) %>% unique() -->
<!-- # ) -->
<!-- #   -->
<!-- # # With networkD3, connection must be provided using id, not using real name like in the links dataframe.. So we need to reformat it. -->
<!-- # links$IDsource <- match(links$source, nodes$name)-1  -->
<!-- # links$IDtarget <- match(links$target, nodes$name)-1 -->
<!-- #   -->
<!-- # # Make the Network -->
<!-- # p <- sankeyNetwork(Links = links, Nodes = nodes, -->
<!-- #               Source = "IDsource", Target = "IDtarget", -->
<!-- #               Value = "value", NodeID = "name",  -->
<!-- #               sinksRight=FALSE) -->

<!-- test <- as.dist(distance_matrix, diag = TRUE) -->
<!-- hc <- hclust(test) -->

<!-- dendroNetwork(hc, height = 500, nodeColour = "darkcyan") -->
<!-- ``` -->



<!-- ### Pathway check -->

<!-- ```{r} -->
<!-- ORA_metabase = function(viperRes, universe, GOs) { #, myGO -->

<!--   entities <-  filter(viperRes, NES != 0) %>% arrange(NES, decrease=T) -->

<!--   symbols_temp <- entities$protein -->
<!--   mapping_symbole_to_entrez_temp <- mapIds(org.Hs.eg.db, symbols_temp, 'ENTREZID', 'SYMBOL') -->
<!--   entities$entrezid <- mapping_symbole_to_entrez_temp[entities$protein] -->

<!--   gene_list <-  entities$entrezid -->
<!--   maps <-  load.ontology(GOs,  type = "gene") #load.ontology("maps", type = "gene") -->

<!--   entities <-  filter(universe, NES != 0) %>% arrange(NES, decrease=T) -->

<!--   symbols_temp <- entities$protein -->
<!--   mapping_symbole_to_entrez_temp <- mapIds(org.Hs.eg.db, symbols_temp, 'ENTREZID', 'SYMBOL') -->
<!--   entities$entrezid <- mapping_symbole_to_entrez_temp[entities$protein] -->

<!--   universe <-  entities$entrezid -->


<!--   er <- metabaser::enrichment(genelist = as.character(gene_list),  -->
<!--                               background.list = as.character(universe),  -->
<!--                               ontology = maps, -->
<!--                               min.r = 2, -->
<!--                               alpha = 1) -->


<!--   er_det <- er %>% mutate(pathway = rownames((er))) %>% -->
<!--     mutate(pathway_proteins = maps[pathway]) %>% -->
<!--     mutate(gene_list = list(gene_list)) %>% -->
<!--     rowwise() %>% -->
<!--     mutate(intersect =  list(intersect(gene_list, pathway_proteins))) %>% -->
<!--     rowwise() %>% -->
<!--     mutate(intersect = list( as.vector( mapIds(org.Hs.eg.db, intersect, 'SYMBOL', 'ENTREZID')))) %>% -->
<!--     mutate(proportion = r/n) %>% -->
<!--     dplyr::select(-pathway_proteins, -gene_list) -->
<!--   return(er_det) -->
<!-- } -->

<!-- ``` -->


<!-- ```{r, message=F, warning=F} -->
<!-- load("results/2021_09_05_12_15_40_TPPactestNA_NAweight_Fstat_adjrev_viper_GSK_rep1.RData") -->
<!-- pathways_universe <- COSMOS_object_fw1$meta_network %>%  -->
<!--   mutate(source = gsub("X", "", source), target = gsub("X", "", target)) -->

<!-- pathways_universe <- data.frame(protein = mapIds(org.Hs.eg.db, unique(c(pathways_universe$source, pathways_universe$target)), -->
<!--                                                  'SYMBOL', 'ENTREZID'), NES = 1) -->


<!-- pathway_input_gathered <- map_dfr(result_list, 'nodes', .id = "networkID") %>%  -->
<!--   dplyr::select(networkID, id) -->


<!-- pathway_enrichment <- pathway_input_gathered %>%  -->
<!--   #nest(c(id))%>%  -->
<!--   group_by(networkID) %>%  -->
<!--   mutate(enrichment = list(ORA_metabase(data.frame(protein = id, NES = 1),  -->
<!--                                    pathways_universe,  -->
<!--                                    "snetws"))) -->


<!-- pathway_enrichment_unnest <- pathway_enrichment %>%  -->
<!--   distinct(networkID, enrichment) %>%  -->
<!--   unnest() %>%  -->
<!--   separate(networkID, into = c("TPP_set", "replicate"), sep = "_", remove = F) -->


<!-- pathway_enrichment_significant <- pathway_enrichment_unnest %>%  -->
<!--   filter(qvalue < 0.05) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- p_pathway_enrichment <- ggplot(pathway_enrichment_significant, aes(y = Zscore, x = reorder(pathway, -Zscore), fill = networkID)) + -->
<!--   geom_bar(stat = "identity") + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   coord_flip() + -->
<!--   facet_wrap(~TPP_set, ncol =4) + -->
<!--   theme_bw(12)+ -->
<!--   labs(y = "Zscore pathway enrichment", x = "pathway", title = "Significant pathways of network nodes") -->
<!-- p_pathway_enrichment -->
<!-- ``` -->

<!-- ```{r} -->
<!-- pathways_all_three <- pathway_enrichment_significant %>%  -->
<!--   filter(TPP_set != "noTPP") %>%  -->
<!--   ungroup() %>%  -->
<!--   distinct(TPP_set, pathway) %>%  -->
<!--   group_by(pathway) %>%  -->
<!--   mutate(number_TPPsets = n()) %>%  -->
<!--   right_join(pathway_enrichment_significant, by = c("TPP_set", "pathway")) %>%  -->
<!--   filter(number_TPPsets ==3) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- knitr::kable(pathways_all_three %>% distinct(pathway)) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- pathways_TPPunique <- pathway_enrichment_significant %>%  -->
<!--    anti_join(filter(pathway_enrichment_significant, TPP_set == "noTPP" ), by = "pathway") %>%  -->
<!--   ungroup() %>%  -->
<!--   distinct(TPP_set, pathway) %>%  -->
<!--   group_by(pathway) %>%  -->
<!--   mutate(number_TPPsets = n(), TPPsets = list(TPP_set)) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- knitr::kable(pathways_TPPunique %>% distinct(pathway, number_TPPsets, TPPsets)) -->
<!-- ``` -->



<!-- ```{r} -->
<!-- interesting_pathwaygenes <- unlist(pathway_enrichment_significant$intersect[pathway_enrichment_significant$pathway == "DNA damage_Checkpoint"]) %>%  -->
<!--   unique() -->

<!-- centralties_interestingpathway <- centralities %>%  -->
<!--   filter(Node %in% interesting_pathwaygenes) -->

<!-- p_centralties_interestingpathway <- ggplot(centralties_interestingpathway, aes(x = reorder(Node, degree), y = degree, fill = networkID)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   coord_flip() + -->
<!--   theme_bw(12)+ -->
<!--   labs(y = "degree centrality", x = "pathway member", title = "Centrality of DNA damage_Checkpoint members in differnet networks") -->
<!-- p_centralties_interestingpathway -->

<!-- ``` -->

<!-- ```{r} -->

<!-- interesting_pathwaygenes <- unlist(pathways_TPPunique$intersect[pathways_TPPunique$pathway == "Apoptosis_Apoptotic nucleus"]) %>%  -->
<!--   unique() -->

<!-- centralties_interestingpathway <- centralities %>%  -->
<!--   filter(Node %in% interesting_pathwaygenes) -->

<!-- p_centralties_interestingpathway <- ggplot(centralties_interestingpathway, aes(x = reorder(Node, degree), y = degree, fill = networkID)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   coord_flip() + -->
<!--   theme_bw(12)+ -->
<!--   labs(y = "degree centrality", x = "pathway member", title = "Centrality of Apoptosis pathway members in differnet networks") -->
<!-- p_centralties_interestingpathway -->
<!-- ``` -->

<!-- ```{r} -->
<!-- interesting_pathwaygenes <- unlist(pathways_all_three$intersect[pathways_all_three$pathway == "DNA damage_Checkpoint"]) %>%  -->
<!--   unique() -->

<!-- centralties_interestingpathway <- centralities %>%  -->
<!--   filter(Node %in% interesting_pathwaygenes) -->

<!-- p_centralties_interestingpathway <- ggplot(centralties_interestingpathway, aes(x = reorder(Node, degree), y = degree, fill = networkID)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + -->
<!--   scale_fill_manual(values = values_networkID) + -->
<!--   coord_flip() + -->
<!--   theme_bw(12)+ -->
<!--   labs(y = "degree centrality", x = "pathway member", title = "Centrality of DNA damage Checkpoint pathway members in differnet networks") -->
<!-- p_centralties_interestingpathway -->
<!-- ``` -->


<!-- ```{r} -->
<!-- interesting_pathwaygenes <- unlist(pathways_all_three$intersect[pathways_all_three$pathway == "DNA damage_Checkpoint"]) -->

<!-- filtered_network <- res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$combined_network %>% -->
<!--   as.data.frame() %>% -->
<!--   filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) -->

<!-- filtered_nodes <- res_2021_08_26_10_59_55_TPPactestNA_NAweight_Fstat_adjrev_viper_COSMOSnew_rep1$nodes %>% -->
<!--   as.data.frame() %>% -->
<!--   filter(id %in% filtered_network$from | id %in% filtered_network$to) -->


<!-- test = visNetwork(filtered_nodes , distinct(filtered_network)) %>% -->
<!--     visGroups(groupname = "TPP_active", color = "seagreen") %>% -->
<!--     visGroups(groupname = "TPP_inactive", color = "palegreen") %>% -->
<!--     visGroups(groupname = "TF_active", color = "darkred") %>% -->
<!--     visGroups(groupname = "TF_inactive", color = "orangered") %>% -->
<!--     visGroups(groupname = "Kinase_active", color = "navy") %>% -->
<!--     visGroups(groupname = "Kinase_inactive", color = "lightskyblue") %>% -->
<!--     visGroups(groupname = "other_active", color = "gray") %>% -->
<!--     visGroups(groupname = "other_inactive", color = "gainsboro") %>% -->
<!--     # standard options -->
<!--     visEdges(arrows = "to") %>% -->
<!--     visLegend() %>% -->
<!--     visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>% -->
<!--     visLayout(randomSeed = 123) -->


<!-- test -->

<!-- ``` -->


<!-- ```{r} -->

<!-- ``` -->



<!-- ```{r} -->
<!-- BiocManager::install("RCy3") -->
<!-- library(RCy3) -->
<!--  cytoscapeVersionInfo() -->
<!--  commandsAPI() -->


<!-- nodes_test <- data.frame(id=c("node 0","node 1","node 2","node 3"), -->
<!--            group=c("A","A","B","B"), # categorical strings -->
<!--            score=as.integer(c(20,10,15,5)), # integers -->
<!--            stringsAsFactors=FALSE) -->
<!-- edges_test <- data.frame(source=c("node 0","node 0","node 0","node 2"), -->
<!--            target=c("node 1","node 2","node 3","node 3"), -->
<!--            interaction=c("inhibits","interacts","activates","interacts"),  # optional -->
<!--            weight=c(5.1,3.0,5.2,9.9), # numeric -->
<!--            stringsAsFactors=FALSE) -->

<!-- createNetworkFromDataFrames(nodes_test,edges_test, title="my first network", collection="DataFrame Example") -->
<!-- ``` -->


<!-- ```{r} -->
<!-- interesting_pathwaygenes <- unlist(pathway_check_result$intersect[pathway_check_result$pathway == "DNA damage_Checkpoint"]) -->

<!-- filtered_network <- res_2021_08_10_09_49_25_TPPcorrKinasesvsTF_viper_COSMOSol$combined_network %>% -->
<!--   as.data.frame() %>% -->
<!--   filter(from %in% interesting_pathwaygenes | to %in% interesting_pathwaygenes) -->

<!-- filtered_nodes <- res_2021_08_10_09_49_25_TPPcorrKinasesvsTF_viper_COSMOSol$nodes %>% -->
<!--   as.data.frame() %>% -->
<!--   filter(id %in% filtered_network$from | id %in% filtered_network$to) -->


<!-- test = visNetwork(filtered_nodes , distinct(filtered_network)) %>% -->
<!--     visGroups(groupname = "TPP_active", color = "seagreen") %>% -->
<!--     visGroups(groupname = "TPP_inactive", color = "palegreen") %>% -->
<!--     visGroups(groupname = "TF_active", color = "darkred") %>% -->
<!--     visGroups(groupname = "TF_inactive", color = "orangered") %>% -->
<!--     visGroups(groupname = "Kinase_active", color = "navy") %>% -->
<!--     visGroups(groupname = "Kinase_inactive", color = "lightskyblue") %>% -->
<!--     visGroups(groupname = "other_active", color = "gray") %>% -->
<!--     visGroups(groupname = "other_inactive", color = "gainsboro") %>% -->
<!--     # standard options -->
<!--     visEdges(arrows = "to") %>% -->
<!--     visLegend() %>% -->
<!--     visOptions(highlightNearest = TRUE, nodesIdSelection = TRUE) %>% -->
<!--     visLayout(randomSeed = 123) -->


<!-- test -->

<!-- ``` -->

<!-- ```{r} -->
<!-- load("old/2021_07_23_22_09_05_TFvsKinases_COSMOSold.RData") -->
<!-- ``` -->

```{r}

```

